<!DOCTYPE html>
<html lang="zh-CN">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <title>生信进展-ATAC数据处理 - 欢迎来到我的空间🎈</title><meta name="Description" content="欢迎来到griedzx的空间"><meta property="og:url" content="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/">
  <meta property="og:site_name" content="欢迎来到我的空间🎈">
  <meta property="og:title" content="生信进展-ATAC数据处理">
  <meta property="og:description" content="实验材料 SRR5748809 : Run Browser : SRA Archive : NCBI (nih.gov) 关于玉米（Zea mays）叶肉细胞（leaf mesophyll）的ATAC-seq数据，用于研究染色质可及性的高">
  <meta property="og:locale" content="zh-CN">
  <meta property="og:type" content="article">
    <meta property="article:section" content="posts">
    <meta property="article:published_time" content="2024-03-09T16:56:43+08:00">
    <meta property="article:modified_time" content="2024-03-09T16:56:43+08:00">
    <meta property="og:image" content="http://griedzx.github.io/logo.png">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="http://griedzx.github.io/logo.png"><meta name="twitter:title" content="生信进展-ATAC数据处理">
<meta name="twitter:description" content="实验材料 SRR5748809 : Run Browser : SRA Archive : NCBI (nih.gov) 关于玉米（Zea mays）叶肉细胞（leaf mesophyll）的ATAC-seq数据，用于研究染色质可及性的高">
      <meta name="twitter:site" content="@xxxx">
<meta name="application-name" content="我的网站">
<meta name="apple-mobile-web-app-title" content="我的网站"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" /><link rel="prev" href="http://griedzx.github.io/molecular-biology/" /><link rel="next" href="http://griedzx.github.io/ngs_visualization/" /><link rel="stylesheet" href="/css/style.min.css"><link rel="preload" href="/lib/fontawesome-free/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"></noscript><link rel="preload" href="/lib/animate/animate.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/animate/animate.min.css"></noscript><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "生信进展-ATAC数据处理",
        "inLanguage": "zh-CN",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "http:\/\/griedzx.github.io\/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86\/"
        },"genre": "posts","wordcount":  5130 ,
        "url": "http:\/\/griedzx.github.io\/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86\/","datePublished": "2024-03-09T16:56:43+08:00","dateModified": "2024-03-09T16:56:43+08:00","publisher": {
            "@type": "Organization",
            "name": ""},"author": {
                "@type": "Person",
                "name": "griedzx"
            },"description": ""
    }
    </script></head>
    <body data-header-desktop="fixed" data-header-mobile="auto"><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('light' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : 'light' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="欢迎来到我的空间🎈">griedzx 的空间<span class="header-title-post">⚽</span></a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/">🤴 文章 </a><a class="menu-item" href="/tags/"> 标签 </a><a class="menu-item" href="/categories/"> 分类 </a><a class="menu-item" href="https://upptime.daizixi.space/" rel="noopener noreffer" target="_blank"> Uptime-Status </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="搜索">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="清空">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                    <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
                </a></div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="欢迎来到我的空间🎈">griedzx 的空间<span class="header-title-post">⚽</span></a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="搜索">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="清空">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        取消
                    </a>
                </div><a class="menu-item" href="/posts/" title="">🤴文章</a><a class="menu-item" href="/tags/" title="">标签</a><a class="menu-item" href="/categories/" title="">分类</a><a class="menu-item" href="https://upptime.daizixi.space/" title="" rel="noopener noreffer" target="_blank">Uptime-Status</a><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
            </a></div>
    </div>
</header><div class="search-dropdown desktop">
        <div id="search-dropdown-desktop"></div>
    </div>
    <div class="search-dropdown mobile">
        <div id="search-dropdown-mobile"></div>
    </div><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
  <meta name="referrer" content="no-referrer-when-downgrade"><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">目录</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animate__animated animate__flipInX">生信进展-ATAC数据处理</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="http://mail.qq.com/cgi-bin/qm_share?t=qm_mailme&amp;email=lOf88frw4aSip6TU5eW69-v5" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw" aria-hidden="true"></i>griedzx</a></span>&nbsp;<span class="post-category">收录于 <a href="/categories/%E5%A4%84%E7%90%86%E5%92%8C%E5%88%86%E6%9E%90%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE/"><i class="far fa-folder fa-fw" aria-hidden="true"></i>处理和分析高通量测序数据</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw" aria-hidden="true"></i>&nbsp;<time datetime="2024-03-09">2024-03-09</time>&nbsp;<i class="fas fa-pencil-alt fa-fw" aria-hidden="true"></i>&nbsp;约 5130 字&nbsp;
                <i class="far fa-clock fa-fw" aria-hidden="true"></i>&nbsp;预计阅读 11 分钟&nbsp;</div>
        </div><div class="details toc" id="toc-static"  data-kept="true">
                <div class="details-summary toc-title">
                    <span>目录</span>
                    <span><i class="details-icon fas fa-angle-right" aria-hidden="true"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#实验材料">实验材料</a></li>
    <li><a href="#实验流程">实验流程</a>
      <ul>
        <li><a href="#软件">软件</a></li>
        <li><a href="#分析步骤">分析步骤</a>
          <ul>
            <li><a href="#数据预处理">数据预处理</a></li>
            <li><a href="#序列比对">序列比对</a></li>
            <li><a href="#peak-calling">Peak calling</a></li>
            <li><a href="#生成bwbigwig文件">生成bw/bigwig文件</a></li>
            <li><a href="#下游分析">下游分析</a>
              <ul>
                <li><a href="#peak注释">Peak注释</a></li>
                <li><a href="#峰基因go富集分析">峰基因GO富集分析</a></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><a href="#实验结果">实验结果</a>
      <ul>
        <li><a href="#数据预处理-1">数据预处理</a></li>
        <li><a href="#序列比对-1">序列比对</a></li>
        <li><a href="#peak-calling-1">peak calling</a></li>
        <li><a href="#下游分析-1">下游分析</a></li>
      </ul>
    </li>
    <li><a href="#实验总结">实验总结</a></li>
    <li><a href="#参考">参考</a></li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h2 id="实验材料">实验材料</h2>
<p><a href="https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&amp;acc=SRR5748809&amp;display=analysis" target="_blank" rel="noopener noreffer ">SRR5748809 : Run Browser : SRA Archive : NCBI (nih.gov)</a></p>
<p>关于玉米（Zea mays）叶肉细胞（leaf mesophyll）的ATAC-seq数据，用于研究染色质可及性的高通量测序数据</p>
<h2 id="实验流程">实验流程</h2>
<h3 id="软件">软件</h3>
<p>SRA-tools <a href="https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/" target="_blank" rel="noopener noreffer ">Index of /sra/sdk (nih.gov)</a></p>
<p>FastQC <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener noreffer ">Babraham Bioinformatics - FastQC A Quality Control tool for High Throughput Sequence Data</a></p>
<p>BWA <a href="https://github.com/lh3/bwa" target="_blank" rel="noopener noreffer ">lh3/bwa: Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment) (github.com)</a></p>
<p>samtools <a href="https://github.com/samtools/samtools" target="_blank" rel="noopener noreffer ">samtools/samtools: Tools (written in C using htslib) for manipulating next-generation sequencing data (github.com)</a></p>
<p>bedtools <a href="https://bedtools.readthedocs.io/en/latest/index.html" target="_blank" rel="noopener noreffer ">bedtools: a powerful toolset for genome arithmetic — bedtools 2.31.0 documentation</a></p>
<p>deeptools <a href="https://bedtools.readthedocs.io/en/latest/index.html" target="_blank" rel="noopener noreffer ">bedtools: a powerful toolset for genome arithmetic — bedtools 2.31.0 documentation</a></p>
<p>R-clusterPro <a href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank" rel="noopener noreffer ">Bioconductor - clusterProfiler</a></p>
<h3 id="分析步骤">分析步骤</h3>
<h4 id="数据预处理">数据预处理</h4>
<p>通过itc平台使用逸夫楼网络环境ssh登录</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">ssh bio_zxdai@172.16.192.8
</span></span></code></pre></td></tr></table>
</div>
</div><p>将所需要软件路径加入环境变量</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">cat &gt;&gt; .bashrc <span class="s">&lt;&lt; EOF
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; PATH=/home/software/sratoolkit.3.1.0-centos_linux64/bin/:$PATH
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; EOF</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>或者将上述若干命令写入脚本，将执行脚本命令加入 <code>.bashrc</code>文件</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 ~<span class="o">]</span>$ cat /home/software/softlist
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/bwa-0.7.17:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/FastQC:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/fastp:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/bowtie2-2.4.5-linux-x86_64:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/hisat2-2.1.0:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/tophat-2.1.1.Linux_x86_64:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/bedtools2/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/samtools-1.9:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/subread-2.0.3-Linux-x86_64/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/stringtie-2.2.1.Linux_x86_64:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/miniconda3/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="c1"># alias samtools=&#34;samtools_0.1.18&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/BatMeth2-master/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nb">alias</span> <span class="nv">Trimmomatic</span><span class="o">=</span><span class="s2">&#34;java -jar /home/software/Trimmomatic-0.39/trimmomatic-0.39.jar&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/STAR-master/source/:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/qualimap_v2.2.1/:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="c1">#PATH=/home/software/bowtie-1.3.1/:$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/Bioinfo/zxzhu/software/BatMeth2/bin:<span class="nv">$PATH</span>
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">cat &gt;&gt; .bashrc <span class="s">&lt;&lt; EOF
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; source /home/software/softlist
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; EOF</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>使用NCBI的sratoolkit下载sra数据</p>
<p>prefetch命令下载</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">prefetch SRR5748809
</span></span></code></pre></td></tr></table>
</div>
</div><p>fastq-dump 工具对 SRA 文件进行 fastq 文件切分</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 data<span class="o">]</span>$ fasterq-dump --split-3 SRR5748809 -O fastq
</span></span><span class="line"><span class="cl">spots <span class="nb">read</span>      : 10,480,210
</span></span><span class="line"><span class="cl">reads <span class="nb">read</span>      : 20,960,420
</span></span><span class="line"><span class="cl">reads written   : 20,960,420
</span></span></code></pre></td></tr></table>
</div>
</div><p>fastqc质量控制</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">fastqc -o ./qc -t <span class="m">4</span> -f fastq ./data/fastq/*
</span></span></code></pre></td></tr></table>
</div>
</div><p>序列剪切（Trimmomatic）</p>
<p>创建Trimmomatic命令别名</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="nb">alias</span> <span class="nv">Trimmomatic</span><span class="o">=</span><span class="s2">&#34;java -jar /home/software/Trimmomatic-0.39/trimmomatic-0.39.jar&#34;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>添加adapter文件</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">cp /home/software/Trimmomatic-0.39/adapters/* ./
</span></span><span class="line"><span class="cl">cat NexteraPE-PE.fa - TruSeq3-PE-2.fa &gt;&gt; adapters.fa
</span></span></code></pre></td></tr></table>
</div>
</div><p>单端测序数据剪切</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">Trimmomatic SE -phred33 -threads <span class="m">3</span> /Bioinfo/bio_2023_2024_2/bio_zxdai/ATAC-seq/data/fastq/SRR5748809.fastq SRR5748809_processed.fastq ILLUMINACLIP:./adapters/adapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:20 MINLEN:36 -trimlog log.txt
</span></span></code></pre></td></tr></table>
</div>
</div><p>命令参数解释</p>
<ul>
<li><code>SE</code>：表示这是单端（Single End）测序数据。如果是双端（Paired End）测序数据，这里应该使用 <code>PE</code>。</li>
<li><code>-phred33</code>：指定质量分数的编码方式。Phred33是最常见的编码方式，用于Illumina数据和较新的Sanger数据。</li>
<li><code>-threads 3</code>：指定使用3个线程进行并行处理。</li>
<li><code>/Bioinfo/bio_2023_2024_2/bio_zxdai/ATAC-seq/data/fastq/SRR5748809.fastq</code>：输入文件的路径，这是需要进行质量控制的原始测序数据。</li>
<li><code>SRR5748809_processed.fastq</code>：输出文件的路径，这是经过质量控制后的测序数据。</li>
<li><code>ILLUMINACLIP:./adapters/adapters.fa:2:30:10</code>：使用IlluminaClip步骤移除测序适配器。<code>./adapters/adapters.fa</code>是适配器序列的文件路径，<code>2:30:10</code>是这个步骤的参数，分别表示种子匹配长度、回报和回报字符串匹配度。</li>
<li><code>LEADING:3</code>：移除读取开始部分质量低于3的碱基。</li>
<li><code>TRAILING:3</code>：移除读取结束部分质量低于3的碱基。</li>
<li><code>SLIDINGWINDOW:4:20</code>：使用滑动窗口剪切，当窗口内的平均质量低于20时剪切。窗口大小为4。</li>
<li><code>MINLEN:36</code>：移除长度低于36的读取。</li>
<li><code>-trimlog log.txt</code>：记录每个读取的剪切信息，输出到 <code>log.txt</code>文件。</li>
</ul>
<h4 id="序列比对">序列比对</h4>
<p>agstat.txt</p>
<p>参考基因组下载</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">wget --quiet  -c https://download.maizegdb.org/Zm-B73-REFERENCE-NAM-5.0/Zm-B73-REFERENCE-NAM-5.0.fa.gz
</span></span><span class="line"><span class="cl"><span class="c1">#解压压缩文件</span>
</span></span><span class="line"><span class="cl">gunzip *gz
</span></span></code></pre></td></tr></table>
</div>
</div><p>检查基因组下载是否完整</p>
<p>使用 <code>md5sum</code>命令来检查文件的完整性。这个命令会计算文件的MD5哈希值，可以将这个值与你从下载页面得到的MD5哈希值进行比较</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">md5sum Zm-B73-REFERENCE-NAM-5.0.fa
</span></span></code></pre></td></tr></table>
</div>
</div><p>BWA建立索引、匹配</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#建立索引</span>
</span></span><span class="line"><span class="cl">nohup bwa index ./ref/Zm*.fa <span class="p">&amp;</span> 
</span></span><span class="line"><span class="cl"><span class="c1">#cd回上一级路径</span>
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ..
</span></span></code></pre></td></tr></table>
</div>
</div><p>使用 <code>bwa index</code> 建立索引，索引文件会以 <code>Zm*.fa</code>所匹配到的表达式为文件名前缀</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#align</span>
</span></span><span class="line"><span class="cl"><span class="c1">#mem algorithm</span>
</span></span><span class="line"><span class="cl">bwa mem ./data/ref/Zm-B73-REFERENCE-NAM-5.0.fa ./data/fastq/SRR5748809.fastq 1&gt;./align/SRR5748809.sam 2&gt;./align/bwa_mem.log <span class="p">&amp;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#ALN algorithm</span>
</span></span><span class="line"><span class="cl">bwa aln -t <span class="m">5</span> ./data/ref/Zm-B73-REFERENCE-NAM-5.0.fa ./data/fastq/SRR5748809.fastq 1&gt;./align/SRR5748809.sai 2&gt;./align/bwa_aln.log <span class="p">&amp;</span>
</span></span><span class="line"><span class="cl">bwa samse ./data/ref/Zm-B73-REFERENCE-NAM-5.0.fa ./align/SRR5748809.sai ./data/fastq/SRR5748809.fastq 1&gt;./align/SRR5748809.aln.sam <span class="p">&amp;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p><code>bwa mem</code>是BWA的默认算法，适用于长度70bp-1Mbp的序列。它比 <code>bwa aln</code>更快，更准确，尤其是对于较长的序列。<code>bwa mem</code>也支持gapped alignment（允许插入和删除），并且可以处理paired-end reads</p>
<ul>
<li><code>-t 多线程并行</code></li>
<li><code>1&gt;./align/SRR5748809.sam</code>:  将标准输出（对齐结果）重定向到SAM文件。</li>
<li><code>2&gt;./align/bwa_mem.log</code>:  将标准错误输出（如错误和警告信息）重定向到日志文件。</li>
</ul>
<p><code>bwa aln</code>是BWA的旧算法，适用于长度up to ~100bp的序列。它使用回溯法（backtracking）进行比对，对于较短的序列效果较好。然而，<code>bwa aln</code>不支持gapped alignment，因此对于包含插入和删除的序列，其比对效果可能不如 <code>bwa mem</code>。推荐使用 <code>bwa mem</code></p>
<p>aln算法会先生成 <code>.sai</code> BWA-backtrack算法的对齐结果文件,然后使用 <code>bwa samse</code>&ndash;BWA的一个模块，将BWA-backtrack算法的对齐结果转换为SAM格式</p>
<p>比对结果的统计</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$!</span> &gt; bwa.pid
</span></span><span class="line"><span class="cl"><span class="nb">wait</span> <span class="k">$(</span>cat bwa.pid<span class="k">)</span> <span class="o">&amp;&amp;</span> samtools view -bS ./align/SRR5748809.sam <span class="p">|</span> samtools flagstat - &gt; ./align/flagstat.txt
</span></span></code></pre></td></tr></table>
</div>
</div><p>将测序数据比对回基因组需要花费一定时间，可以使用上述命令或者写成一个脚本以连贯执行命令</p>
<ul>
<li><code>-b</code>: samtools将输出转换为BAM格式</li>
<li><code>-S</code>: 指定samtools输入文件SAM格式</li>
<li><code>-</code>: 表示从标准输入（stdin）读取数据，数据来自于前一个命令的输出</li>
</ul>
<h4 id="peak-calling">Peak calling</h4>
<p>筛选比对结果</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="s2">&#34;SRR5748809&#34;</span>
</span></span><span class="line"><span class="cl"><span class="c1">#去除PCR重复</span>
</span></span><span class="line"><span class="cl"><span class="nv">$picard</span> MarkDuplicates -I ../align/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.bam 
</span></span><span class="line"><span class="cl">    -O<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.rmdup.bam -M <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.rmdup.metric --REMOVE_DUPLICATES <span class="nb">true</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#去除低质量reads(-q 30)以及未必对到同一条染色体(-f 2)的数据</span>
</span></span><span class="line"><span class="cl"><span class="nv">$samtools</span> view -h -f <span class="m">2</span> -q <span class="m">30</span> <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.rmdup.bam <span class="p">|</span> grep -v chrM <span class="p">|</span> <span class="nv">$samtools</span> sort -O bam -@ <span class="m">20</span> -o - &gt;<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.last.bam
</span></span><span class="line"><span class="cl">bedtools bamtobed -i <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.last.bam &gt;<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.bed
</span></span></code></pre></td></tr></table>
</div>
</div><p>call peak</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">mkdir peaks 
</span></span><span class="line"><span class="cl">macs2 callpeak -t ./align/*bed -g <span class="m">14300719022</span> --nomodel --shift -100 --extsize <span class="m">200</span> -n SRR5748809 --outdir ./peaks
</span></span></code></pre></td></tr></table>
</div>
</div><p>参数的解析：</p>
<ul>
<li><code>-t</code>：指定处理组的文件，这里是 BED 格式的文件</li>
<li><code>-g</code>：指定基因组的大小</li>
<li><code>--nomodel</code>：这个选项告诉 MACS2 不要构建移位模型，这对于 ATAC-seq 数据是必要的，因为 ATAC-seq 数据的片段大小不代表转录因子结合位点。</li>
<li><code>--shift</code>：移动读取的中心位置。对于 ATAC-seq 数据，通常会设置为 -100，表示向 5&rsquo; 方向移动 100 bp</li>
<li><code>--extsize</code>：设置读取的延伸大小。对于 ATAC-seq 数据，通常会设置为 200，表示每个读取都会向 3&rsquo; 方向延伸 200 bp</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-text" data-lang="text"><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 tag size is determined as 51 bps 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 tag size = 51.0 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1  total tags in treatment: 27376162 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 user defined the maximum tags... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1  tags after filtering in treatment: 15521232 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1  Redundant rate of treatment: 0.43 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 finished! 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Build Peak Model... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Skipped... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Use 200 as fragment length 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Sequencing ends will be shifted towards 5&#39; by 100 bp(s) 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #3 Call peaks... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #3 Pre-compute pvalue-qvalue table... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:16: #3 Call peaks for each chromosome... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:56: #4 Write output xls file... SRR5748809_peaks.xls 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:56: #4 Write peak in narrowPeak format file... SRR5748809_peaks.narrowPeak 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:57: #4 Write summits bed file... SRR5748809_summits.bed 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:57: Done! 
</span></span></code></pre></td></tr></table>
</div>
</div><p>不使用nomodel参数可以获得model_plot.r，用于绘制片段大小的分布图，反映了开放染色质的区域</p>
<h4 id="生成bwbigwig文件">生成bw/bigwig文件</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png"
        title="1712560424603" width="551" height="435" /></p>
<p>通过路径 bam -&gt; bedGraph -&gt; bigwig/bw</p>
<p>获取 <code>chrom.sizes</code> 文件，其中包含了每个染色体的名称和大小</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">samtools faidx data/ref/*fa
</span></span><span class="line"><span class="cl">cut -f1,2 data/ref/*.fa.fai &gt; data/ref/chrom.sizes
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">bedtools genomecov -bg -i peaks/*summits.bed -g data/ref/chrom.sizes &gt; peaks/SRR5748809.bedGraph
</span></span><span class="line"><span class="cl">bedGraphToBigWig peaks/SRR5748809.bedGraph data/ref/chrom.sizes peaks/SRR5748809.bw
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 ATAC-seq<span class="o">]</span>$ ll peaks
</span></span><span class="line"><span class="cl">total <span class="m">12524</span>
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">1288417</span> Apr  <span class="m">8</span> 15:22 SRR5748809.bedGraph
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai  <span class="m">559174</span> Apr  <span class="m">8</span> 15:22 SRR5748809.bw
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai   <span class="m">25600</span> Apr  <span class="m">8</span> 14:33 SRR5748809_model.pdf
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai   <span class="m">99228</span> Mar <span class="m">28</span> 15:47 SRR5748809_model.r
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">3919519</span> Mar <span class="m">28</span> 15:49 SRR5748809_peaks.narrowPeak
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">4257998</span> Mar <span class="m">28</span> 15:49 SRR5748809_peaks.xls
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">2660699</span> Mar <span class="m">28</span> 15:49 SRR5748809_summits.bed
</span></span></code></pre></td></tr></table>
</div>
</div><p>上游分析结束</p>
<h4 id="下游分析">下游分析</h4>
<h5 id="peak注释">Peak注释</h5>
<p>只考虑玉米染色体1-10</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-R" data-lang="R"><span class="line"><span class="cl"><span class="c1">#BiocManager::install(&#34;GenomicFeatures&#34;)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#BiocManager::install(&#34;ChIPseeker&#34;)</span>
</span></span><span class="line"><span class="cl"><span class="nf">install.packages</span><span class="p">(</span><span class="s">&#34;ggupset&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">ChIPseeker</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#install.packages(&#34;RMariaDB&#34;)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">RMariaDB</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">GenomicFeatures</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">ggimage</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">maize_TxDb</span> <span class="o">&lt;-</span> <span class="nf">makeTxDbFromGFF</span><span class="p">(</span><span class="s">&#34;D:/seq_project/seq_project/work_data/Zma_B73_V4.gff3&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">wd</span> <span class="o">&lt;-</span> <span class="s">&#34;F:/course/生物信息学进展/ATAC-seq/ATAC&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nf">setwd</span><span class="p">(</span><span class="n">wd</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">tissue</span> <span class="o">&lt;-</span> <span class="s">&#34;Zea_mays&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">bed_file</span> <span class="o">&lt;-</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="s">&#34;SRR5748809_summits.bed&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">bed_name</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;_summits.*&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">))</span>
</span></span><span class="line"><span class="cl"><span class="nf">names</span><span class="p">(</span><span class="n">bed_file</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">bed_name</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">bed</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">bed_file</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">bed</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">bed</span><span class="p">,</span> <span class="n">bed</span><span class="o">$</span><span class="n">V1</span> <span class="o">%in%</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;chr&#34;</span><span class="p">,</span> <span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">))</span> <span class="o">%&gt;%</span>
</span></span><span class="line"><span class="cl">        <span class="nf">mutate</span><span class="p">(</span><span class="n">V1</span> <span class="o">=</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;chr&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">V1</span><span class="p">))</span> 
</span></span><span class="line"><span class="cl"><span class="nf">write.table</span><span class="p">(</span><span class="n">bed</span><span class="p">,</span> <span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">bed_name</span><span class="p">,</span> <span class="s">&#34;.bed&#34;</span><span class="p">)),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">bed_file</span> <span class="o">&lt;-</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">bed_name</span><span class="p">,</span> <span class="s">&#34;.bed&#34;</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">peak</span> <span class="o">&lt;-</span> <span class="nf">readPeakFile</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">bed_name</span><span class="p">,</span> <span class="s">&#34;.bed&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="c1"># # 将染色体名称的格式改为 &#34;chr&#34; 开头</span>
</span></span><span class="line"><span class="cl"><span class="c1"># seqlevelsStyle(maize_TxDb) &lt;- &#34;UCSC&#34;</span>
</span></span><span class="line"><span class="cl"><span class="c1"># # 检查改变后的染色体名称</span>
</span></span><span class="line"><span class="cl"><span class="c1"># seqlevels(maize_TxDb)s</span>
</span></span><span class="line"><span class="cl"><span class="n">peakAnno</span> <span class="o">&lt;-</span> <span class="nf">annotatePeak</span><span class="p">(</span><span class="n">peak</span><span class="p">,</span> <span class="n">tssRegion</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span> <span class="m">1000</span><span class="p">),</span> <span class="n">TxDb</span><span class="o">=</span><span class="n">maize_TxDb</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_pie&#34;</span><span class="p">,</span> <span class="s">&#34;.pdf&#34;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="nf">plotAnnoPie</span><span class="p">(</span><span class="n">peakAnno</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_upset&#34;</span><span class="p">,</span> <span class="s">&#34;.pdf&#34;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">upsetplot</span><span class="p">(</span><span class="n">peakAnno</span><span class="p">,</span> <span class="n">vennpie</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;genebody_PeakProf&#34;</span><span class="p">,</span> <span class="s">&#34;.pdf&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="n">p_1</span> <span class="o">&lt;-</span> <span class="nf">plotPeakProf2</span><span class="p">(</span><span class="n">bed_file</span><span class="p">,</span> <span class="n">upstream</span> <span class="o">=</span> <span class="m">1000</span><span class="p">,</span> <span class="n">downstream</span> <span class="o">=</span> <span class="m">1000</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">        <span class="n">conf</span> <span class="o">=</span> <span class="m">0.95</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="s">&#34;gene&#34;</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;body&#34;</span><span class="p">,</span> <span class="n">nbin</span> <span class="o">=</span> <span class="m">800</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">        <span class="n">TxDb</span> <span class="o">=</span> <span class="n">maize_TxDb</span><span class="p">,</span> <span class="n">facet</span> <span class="o">=</span> <span class="s">&#34;row&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">p_1</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>TSS/gene body 位点peak plot展示</p>
<p>deeptools</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">computeMatrix scale-regions -p <span class="m">12</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -S ./peaks/SRR5748809.bw <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -R ./peaks/SRR5748809_summits.bed <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -b <span class="m">100000</span> -a <span class="m">100000</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o ./plot/SRR5748809.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --binSize <span class="m">10</span> --missingDataAsZero <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --skipZeros <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --outFileNameMatrix ./plot/SRR5748809_matrix.tab
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">plotHeatmap -m ./plot/SRR5748809.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -out ./plot/SRR5748809.pdf <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --dpi <span class="m">300</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --colorMap YlGnBu  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --missingDataColor <span class="s2">&#34;#FFF6EB&#34;</span><span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --heatmapHeight <span class="m">21</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --startLabel <span class="s2">&#34;TSS&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --endLabel <span class="s2">&#34;TTS&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --regionsLabel <span class="s2">&#34;gene&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --legendLocation <span class="s2">&#34;none&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --plotTitle <span class="s2">&#34;Zea mays&#34;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><img src="image/index/1713284604918.png" alt="ATAC-seq峰值图&热图" height=800>
<h5 id="峰基因go富集分析">峰基因GO富集分析</h5>
<p>Gene Ontology</p>
<p>使用 <code>ChIPseeker</code>包将峰值注释到最近的基因，GO富集分析</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-R" data-lang="R"><span class="line"><span class="cl"><span class="n">peakAnno</span> <span class="o">&lt;-</span> <span class="nf">annotatePeak</span><span class="p">(</span><span class="n">peak</span><span class="p">,</span> <span class="n">tssRegion</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span> <span class="m">1000</span><span class="p">),</span> <span class="n">TxDb</span><span class="o">=</span><span class="n">maize_TxDb</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#annotated peaks to gene and clusterProfiler</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">AnnotationHub</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">hub</span> <span class="o">&lt;-</span> <span class="nf">AnnotationHub</span><span class="p">()</span>
</span></span><span class="line"><span class="cl"><span class="n">maize</span> <span class="o">&lt;-</span> <span class="n">hub[[</span><span class="s">&#34;AH114308&#34;</span><span class="n">]]</span>
</span></span><span class="line"><span class="cl"><span class="n">entrez_gene_id</span> <span class="o">&lt;-</span> <span class="nf">read.csv</span><span class="p">(</span><span class="s">&#34;D:/motif_project/16.tss_maxmotif_GO/data/entrez_Zm_gene_v4.csv&#34;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">gene_ids</span> <span class="o">&lt;-</span> <span class="nf">unique</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">peakAnno</span><span class="p">)</span><span class="o">$</span><span class="n">geneId</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">as.data.frame</span><span class="p">()</span>
</span></span><span class="line"><span class="cl"><span class="nf">colnames</span><span class="p">(</span><span class="n">gene_ids</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="s">&#34;geneId&#34;</span>
</span></span><span class="line"><span class="cl"><span class="n">genes</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">gene_ids</span><span class="p">,</span> <span class="n">entrez_gene_id</span><span class="p">,</span> <span class="n">by.x</span> <span class="o">=</span> <span class="s">&#34;geneId&#34;</span><span class="p">,</span> <span class="n">by.y</span> <span class="o">=</span> <span class="s">&#34;gene_name&#34;</span><span class="p">,</span> <span class="n">all.x</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> <span class="o">%&gt;%</span>
</span></span><span class="line"><span class="cl">        <span class="nf">na.omit</span><span class="p">()</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">ego</span> <span class="o">&lt;-</span> <span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span>         <span class="o">=</span> <span class="n">genes</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">OrgDb</span>        <span class="o">=</span> <span class="n">maize</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">keyType</span>      <span class="o">=</span> <span class="s">&#34;ENTREZID&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">pAdjustMethod</span> <span class="o">=</span> <span class="s">&#34;BH&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">pvalueCutoff</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">qvalueCutoff</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">readable</span>     <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> 
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">head</span><span class="p">(</span><span class="n">ego</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">write.table</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span> <span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_GO.tsv&#34;</span><span class="p">)),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_GO_barplot.pdf&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="n">dotp</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">        <span class="nf">aes</span><span class="p">(</span><span class="n">GeneRatio</span><span class="p">,</span> <span class="nf">fct_reorder</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span> <span class="n">GeneRatio</span><span class="p">)))</span> <span class="o">+</span> 
</span></span><span class="line"><span class="cl">        <span class="nf">geom_segment</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">xend</span><span class="o">=</span><span class="m">0</span><span class="p">,</span> <span class="n">yend</span> <span class="o">=</span> <span class="n">Description</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">p.adjust</span><span class="p">,</span> <span class="n">size</span> <span class="o">=</span> <span class="n">Count</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">scale_color_viridis_c</span><span class="p">(</span><span class="n">guide</span><span class="o">=</span><span class="nf">guide_colorbar</span><span class="p">(</span><span class="n">reverse</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">scale_size_continuous</span><span class="p">(</span><span class="n">range</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span> <span class="m">10</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">theme_minimal</span><span class="p">()</span> <span class="o">+</span> 
</span></span><span class="line"><span class="cl">        <span class="nf">ylab</span><span class="p">(</span><span class="kc">NULL</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">dotp</span><span class="p">)</span> 
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="实验结果">实验结果</h2>
<h3 id="数据预处理-1">数据预处理</h3>
<p>原始SRR5748809 QC</p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png" title="1710343095852" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png" data-sub-html="<h2>原始数据FastQC summary</h2><p>1710343095852</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png" width="470" height="699" />
    </a><figcaption class="image-caption">原始数据FastQC summary</figcaption>
    </figure></p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png" title="1710343119909" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png" data-sub-html="<h2>原始数据 Bsic Statistics</h2><p>1710343119909</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png" width="1072" height="404" />
    </a><figcaption class="image-caption">原始数据 Bsic Statistics</figcaption>
    </figure></p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png" title="1710343138678" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png" data-sub-html="<h2>原始数据质量箱线图分布</h2><p>1710343138678</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png" width="1175" height="806" />
    </a><figcaption class="image-caption">原始数据质量箱线图分布</figcaption>
    </figure></p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png" title="1710343385005" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png" data-sub-html="<h2>原始数据无过表达序列和接头</h2><p>1710343385005</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png" width="1243" height="999" />
    </a><figcaption class="image-caption">原始数据无过表达序列和接头</figcaption>
    </figure></p>
<p>使用上述Trimmomatic命令剪切后，质量没有之前好</p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png" title="1710343706916" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png" data-sub-html="<h2>剪切数据质量summary</h2><p>1710343706916</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png" width="344" height="486" />
    </a><figcaption class="image-caption">剪切数据质量summary</figcaption>
    </figure></p>
<p>其中per base sequence content有些许改变，似乎可以通过 <code>HEADCROP:14</code>参数删除前14个碱基，但序列本身不是很长，便没有尝试</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png"
        title="1710343744796" width="1119" height="796" /></p>
<h3 id="序列比对-1">序列比对</h3>
<p><a href="/2024_spring_ATAC/flagstat.txt" rel="">序列比对数据结果</a></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 align<span class="o">]</span>$ cat flagstat.txt
</span></span><span class="line"><span class="cl"><span class="m">27933221</span> + <span class="m">0</span> in total <span class="o">(</span>QC-passed reads + QC-failed reads<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> secondary
</span></span><span class="line"><span class="cl"><span class="m">34</span> + <span class="m">0</span> supplementary
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> duplicates
</span></span><span class="line"><span class="cl"><span class="m">27376162</span> + <span class="m">0</span> mapped <span class="o">(</span>98.01% : N/A<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> paired in sequencing
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> read1
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> read2
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> properly paired <span class="o">(</span>N/A : N/A<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> with itself and mate mapped
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> singletons <span class="o">(</span>N/A : N/A<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> with mate mapped to a different chr
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> with mate mapped to a different chr <span class="o">(</span>mapQ&gt;<span class="o">=</span>5<span class="o">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="peak-calling-1">peak calling</h3>
<p>peak calling结果分析</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 tag size is determined as 51 bps </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 tag size = 51.0 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1  total tags in treatment: 27376162 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 user defined the maximum tags... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1  tags after filtering in treatment: 15521232 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1  Redundant rate of treatment: 0.43 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 finished! </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Build Peak Model... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Skipped... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Use 200 as fragment length </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Sequencing ends will be shifted towards 5&#39; by 100 bp(s) </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#3 Call peaks... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#3 Pre-compute pvalue-qvalue table... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:16: <span class="c1">#3 Call peaks for each chromosome... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:56: <span class="c1">#4 Write output xls file... SRR5748809_peaks.xls </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:56: <span class="c1">#4 Write peak in narrowPeak format file... SRR5748809_peaks.narrowPeak </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:57: <span class="c1">#4 Write summits bed file... SRR5748809_summits.bed </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:57: Done! 
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li><code>tag size is determined as 51 bps</code>：reads的大小为51个碱基对</li>
<li><code>filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s)</code>：同一位置和同一链上的冗余标签被过滤掉</li>
<li><code>tags after filtering in treatment: 15521232</code>：过滤后，处理的样本中有15521232个tags</li>
<li><code>Redundant rate of treatment: 0.43</code>：处理的样本中的冗余率为0.43</li>
</ul>
<p><a href="/2024_spring_ATAC/SRR5748809_model.pdf" rel="">SRR5748809_model.pdf</a></p>
<p>反映了开放染色质的区域，这个图显示出多个峰值，这些峰值对应于不同的核小体状态</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png"
        title="1712559028672" width="796" height="698" /></p>
<p>交叉相关性（Cross-correlation）图，x轴代表着滞后（lag），也就是读取的移位量，y轴代表着交叉相关性，也就是在给定的滞后下，正向和反向读取的相关性，峰值的位置表示了最佳的滞后，也就是最佳的片段大小</p>
<p>只有一个明显的峰值，这表示大部分的读取都集中在结合位点附近</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png"
        title="1712559047974" width="793" height="787" /></p>
<h3 id="下游分析-1">下游分析</h3>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png"
        title="1712728314927" width="999" height="308" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png"
        title="1712728353944" width="835" height="832" /></p>
<p>GO富集分析</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png"
        title="1713284842478" width="740" height="654" /></p>
<h2 id="实验总结">实验总结</h2>
<p>原始测序数据不错，使用FastQC质量控制查看一下，保证</p>
<ol>
<li>序列质量合格</li>
<li>没有过表达的primer adaptor</li>
<li>序列长度不是很短 即可</li>
</ol>
<p>序列比对数据结果分析</p>
<ul>
<li><code>27933221 + 0 in total (QC-passed reads + QC-failed reads)</code>: 总共有27933221个reads，其中没有QC失败的reads。测序数据的质量应该是相当好的</li>
<li><code>0 + 0 secondary</code>: 没有secondary（次要的）对齐，这些通常是由于reads可以对齐到多个位置。</li>
<li><code>34 + 0 supplementary</code>: 有34个补充的对齐，这些通常是由于reads可以对齐到多个位置。</li>
<li><code>27376162 + 0 mapped (98.01% : N/A)</code>: 有27376162个reads被映射到参考序列，占总reads的98.01%。非常高的映射率，说明大部分reads都能找到参考序列上的对应位置</li>
<li><code>0 + 0 paired in sequencing</code>、<code>0 + 0 read1</code>、<code>0 + 0 read2</code>、<code>0 + 0 properly paired (N/A : N/A)</code>、<code>0 + 0 with itself and mate mapped</code>、<code>0 + 0 singletons (N/A : N/A)</code>、<code>0 + 0 with mate mapped to a different chr</code>、<code>0 + 0 with mate mapped to a different chr (mapQ&gt;=5)</code>: 这些统计信息都是关于配对的reads的，但在这个数据集中，所有的reads都是单端的，所以这些统计信息都是0</li>
</ul>
<p>基因富集分析结论</p>
<p>GO富集分析结果显示了&quot;protein-containing complex binding&quot;、&ldquo;cytoskeletal motor activity&quot;和&quot;microtubule motor activity&quot;这三个GO项。这可能意味着在叶肉细胞中，有一些基因的调控区域是开放的，这些基因可能参与到蛋白质复合物的结合和细胞骨架的运动中。这些过程可能与叶肉细胞的一些重要功能有关，例如光合作用、物质运输等</p>
<ol>
<li>&ldquo;protein-containing complex binding&rdquo;：涉及到的基因编码的蛋白质能够与包含多种蛋白质的复合物结合。这可能意味着ATAC峰值附近基因可能参与到一些需要多种蛋白质共同作用的生物过程中，例如信号转导、基因转录调控等。</li>
<li>&ldquo;cytoskeletal motor activity&quot;和&quot;microtubule motor activity&rdquo;：这两个GO项都与细胞骨架和微管运动有关，参与到许多重要的细胞过程中，例如细胞分裂、物质运输等。</li>
</ol>
<h2 id="参考">参考</h2>
<p><a href="https://www.cnblogs.com/djx571/p/9493934.html" target="_blank" rel="noopener noreffer ">fasta/fastq格式解读 - 发那个太丢人 - 博客园 (cnblogs.com)</a></p>
<p><a href="https://cloud.tencent.com/developer/article/1772411" target="_blank" rel="noopener noreffer ">生信软件 | Sratools (操作SRA文件)-腾讯云开发者社区-腾讯云 (tencent.com)</a></p>
<p><a href="https://www.jianshu.com/p/bdfa8f7e5a61" target="_blank" rel="noopener noreffer ">转录组分析——三、sra转换成fastq - 简书 (jianshu.com)</a></p>
<p><a href="https://www.jianshu.com/p/5c97a34cc1ad" target="_blank" rel="noopener noreffer ">都8102年了，还用fastq-dump，快换fasterq-dump吧 - 简书 (jianshu.com)</a></p>
<p><a href="https://zhuanlan.zhihu.com/p/642000061" target="_blank" rel="noopener noreffer ">转录组之序列剪切（Trimmomatic）[学习笔记通俗易懂版] - 知乎 (zhihu.com)</a></p>
<p><a href="https://zhuanlan.zhihu.com/p/415718382" target="_blank" rel="noopener noreffer ">「与国同庆，万字长文」ATAC-seq实战教程：从SRA数据下载到高分辨率论文主图绘制 - 知乎 (zhihu.com)</a></p>
</div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>更新于 2024-03-09</span>
            </div></div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/index.md" target="_blank">阅读原始文档</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="分享到 Twitter" data-sharer="twitter" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="生信进展-ATAC数据处理" data-via="xxxx"><i class="fab fa-twitter fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 Facebook" data-sharer="facebook" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/"><i class="fab fa-facebook-square fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 Hacker News" data-sharer="hackernews" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="生信进展-ATAC数据处理"><i class="fab fa-hacker-news fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 Line" data-sharer="line" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="生信进展-ATAC数据处理"><i data-svg-src="/lib/simple-icons/icons/line.min.svg" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 微博" data-sharer="weibo" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="生信进展-ATAC数据处理"><i class="fab fa-weibo fa-fw" aria-hidden="true"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">返回</a></span>&nbsp;|&nbsp;<span><a href="/">主页</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/molecular-biology/" class="prev" rel="prev" title="现代分子生物"><i class="fas fa-angle-left fa-fw" aria-hidden="true"></i>现代分子生物</a>
            <a href="/ngs_visualization/" class="next" rel="next" title="NGS可视化">NGS可视化<i class="fas fa-angle-right fa-fw" aria-hidden="true"></i></a></div>
</div>
<div id="comments"></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">由 <a href="https://gohugo.io/" target="_blank" rel="noopener noreffer" title="Hugo 0.125.4">Hugo</a> 强力驱动 | 主题 - <a href="https://github.com/dillonzq/LoveIt" target="_blank" rel="noopener noreffer" title="LoveIt 0.2.11"><i class="far fa-kiss-wink-heart fa-fw" aria-hidden="true"></i> LoveIt</a>
                </div><div class="footer-line" itemscope itemtype="http://schema.org/CreativeWork"><i class="far fa-copyright fa-fw" aria-hidden="true"></i><span itemprop="copyrightYear">2023 - 2024</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="http://mail.qq.com/cgi-bin/qm_share?t=qm_mailme&amp;email=lOf88frw4aSip6TU5eW69-v5" target="_blank">griedzx</a></span>&nbsp;|&nbsp;<span class="license"><a rel="license external nofollow noopener noreffer" href="https://creativecommons.org/licenses/by-nc/4.0/" target="_blank">CC BY-NC 4.0</a></span></div>
        </div>

        
        <div class="busuanzi-footer">
        <span id="busuanzi_container_site_pv">
            本站总访问量<span id="busuanzi_value_site_pv"></span>次
        </span>
        <span id="busuanzi_container_site_uv">
            本站访客数<span id="busuanzi_value_site_uv"></span>人次
        </span>
        </div></footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="回到顶部">
                <i class="fas fa-arrow-up fa-fw" aria-hidden="true"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="查看评论">
                <i class="fas fa-comment fa-fw" aria-hidden="true"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/cookieconsent/cookieconsent.min.css"><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.stemmer.support.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.zh.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/mhchem.min.js"></script><script type="text/javascript" src="/lib/cookieconsent/cookieconsent.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"复制到剪贴板","maxShownLines":50},"comment":{},"cookieconsent":{"content":{"dismiss":"同意","link":"了解更多","message":"本网站使用 Cookies 来改善您的浏览体验."},"enable":true,"palette":{"button":{"background":"#f0f0f0"},"popup":{"background":"#1aa3ff"}},"theme":"edgeless"},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":true,"left":"\\begin{equation}","right":"\\end{equation}"},{"display":true,"left":"\\begin{equation*}","right":"\\end{equation*}"},{"display":true,"left":"\\begin{align}","right":"\\end{align}"},{"display":true,"left":"\\begin{align*}","right":"\\end{align*}"},{"display":true,"left":"\\begin{alignat}","right":"\\end{alignat}"},{"display":true,"left":"\\begin{alignat*}","right":"\\end{alignat*}"},{"display":true,"left":"\\begin{gather}","right":"\\end{gather}"},{"display":true,"left":"\\begin{CD}","right":"\\end{CD}"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"highlightTag":"em","lunrIndexURL":"/index.json","lunrLanguageCode":"zh","lunrSegmentitURL":"/lib/lunr/lunr.segmentit.js","maxResultLength":10,"noResultsFound":"没有找到结果","snippetLength":50,"type":"lunr"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
