<!DOCTYPE html>
<html lang="zh-CN">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <title>ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç† - æ¬¢è¿æ¥åˆ°æˆ‘çš„ç©ºé—´ğŸˆ</title><meta name="Description" content="æ¬¢è¿æ¥åˆ°griedzxçš„ç©ºé—´"><meta property="og:url" content="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/">
  <meta property="og:site_name" content="æ¬¢è¿æ¥åˆ°æˆ‘çš„ç©ºé—´ğŸˆ">
  <meta property="og:title" content="ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†">
  <meta property="og:description" content="å®éªŒææ–™ SRR5748809 : Run Browser : SRA Archive : NCBI (nih.gov) å…³äºç‰ç±³ï¼ˆZea maysï¼‰å¶è‚‰ç»†èƒï¼ˆleaf mesophyllï¼‰çš„ATAC-seqæ•°æ®ï¼Œç”¨äºç ”ç©¶æŸ“è‰²è´¨å¯åŠæ€§çš„é«˜">
  <meta property="og:locale" content="zh-CN">
  <meta property="og:type" content="article">
    <meta property="article:section" content="posts">
    <meta property="article:published_time" content="2024-03-09T16:56:43+08:00">
    <meta property="article:modified_time" content="2024-03-09T16:56:43+08:00">
    <meta property="og:image" content="http://griedzx.github.io/logo.png">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="http://griedzx.github.io/logo.png"><meta name="twitter:title" content="ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†">
<meta name="twitter:description" content="å®éªŒææ–™ SRR5748809 : Run Browser : SRA Archive : NCBI (nih.gov) å…³äºç‰ç±³ï¼ˆZea maysï¼‰å¶è‚‰ç»†èƒï¼ˆleaf mesophyllï¼‰çš„ATAC-seqæ•°æ®ï¼Œç”¨äºç ”ç©¶æŸ“è‰²è´¨å¯åŠæ€§çš„é«˜">
      <meta name="twitter:site" content="@xxxx">
<meta name="application-name" content="æˆ‘çš„ç½‘ç«™">
<meta name="apple-mobile-web-app-title" content="æˆ‘çš„ç½‘ç«™"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" /><link rel="prev" href="http://griedzx.github.io/molecular-biology/" /><link rel="next" href="http://griedzx.github.io/ngs_visualization/" /><link rel="stylesheet" href="/css/style.min.css"><link rel="preload" href="/lib/fontawesome-free/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"></noscript><link rel="preload" href="/lib/animate/animate.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/animate/animate.min.css"></noscript><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†",
        "inLanguage": "zh-CN",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "http:\/\/griedzx.github.io\/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86\/"
        },"genre": "posts","wordcount":  5130 ,
        "url": "http:\/\/griedzx.github.io\/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86\/","datePublished": "2024-03-09T16:56:43+08:00","dateModified": "2024-03-09T16:56:43+08:00","publisher": {
            "@type": "Organization",
            "name": ""},"author": {
                "@type": "Person",
                "name": "griedzx"
            },"description": ""
    }
    </script></head>
    <body data-header-desktop="fixed" data-header-mobile="auto"><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('light' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : 'light' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="æ¬¢è¿æ¥åˆ°æˆ‘çš„ç©ºé—´ğŸˆ">griedzx çš„ç©ºé—´<span class="header-title-post">âš½</span></a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/">ğŸ¤´ æ–‡ç«  </a><a class="menu-item" href="/tags/"> æ ‡ç­¾ </a><a class="menu-item" href="/categories/"> åˆ†ç±» </a><a class="menu-item" href="https://upptime.daizixi.space/" rel="noopener noreffer" target="_blank"> Uptime-Status </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="æœç´¢æ–‡ç« æ ‡é¢˜æˆ–å†…å®¹..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="æœç´¢">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="æ¸…ç©º">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="åˆ‡æ¢ä¸»é¢˜">
                    <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
                </a></div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="æ¬¢è¿æ¥åˆ°æˆ‘çš„ç©ºé—´ğŸˆ">griedzx çš„ç©ºé—´<span class="header-title-post">âš½</span></a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="æœç´¢æ–‡ç« æ ‡é¢˜æˆ–å†…å®¹..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="æœç´¢">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="æ¸…ç©º">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        å–æ¶ˆ
                    </a>
                </div><a class="menu-item" href="/posts/" title="">ğŸ¤´æ–‡ç« </a><a class="menu-item" href="/tags/" title="">æ ‡ç­¾</a><a class="menu-item" href="/categories/" title="">åˆ†ç±»</a><a class="menu-item" href="https://upptime.daizixi.space/" title="" rel="noopener noreffer" target="_blank">Uptime-Status</a><a href="javascript:void(0);" class="menu-item theme-switch" title="åˆ‡æ¢ä¸»é¢˜">
                <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
            </a></div>
    </div>
</header><div class="search-dropdown desktop">
        <div id="search-dropdown-desktop"></div>
    </div>
    <div class="search-dropdown mobile">
        <div id="search-dropdown-mobile"></div>
    </div><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
  <meta name="referrer" content="no-referrer-when-downgrade"><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">ç›®å½•</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animate__animated animate__flipInX">ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="http://mail.qq.com/cgi-bin/qm_share?t=qm_mailme&amp;email=lOf88frw4aSip6TU5eW69-v5" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw" aria-hidden="true"></i>griedzx</a></span>&nbsp;<span class="post-category">æ”¶å½•äº <a href="/categories/%E5%A4%84%E7%90%86%E5%92%8C%E5%88%86%E6%9E%90%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE/"><i class="far fa-folder fa-fw" aria-hidden="true"></i>å¤„ç†å’Œåˆ†æé«˜é€šé‡æµ‹åºæ•°æ®</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw" aria-hidden="true"></i>&nbsp;<time datetime="2024-03-09">2024-03-09</time>&nbsp;<i class="fas fa-pencil-alt fa-fw" aria-hidden="true"></i>&nbsp;çº¦ 5130 å­—&nbsp;
                <i class="far fa-clock fa-fw" aria-hidden="true"></i>&nbsp;é¢„è®¡é˜…è¯» 11 åˆ†é’Ÿ&nbsp;</div>
        </div><div class="details toc" id="toc-static"  data-kept="true">
                <div class="details-summary toc-title">
                    <span>ç›®å½•</span>
                    <span><i class="details-icon fas fa-angle-right" aria-hidden="true"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#å®éªŒææ–™">å®éªŒææ–™</a></li>
    <li><a href="#å®éªŒæµç¨‹">å®éªŒæµç¨‹</a>
      <ul>
        <li><a href="#è½¯ä»¶">è½¯ä»¶</a></li>
        <li><a href="#åˆ†ææ­¥éª¤">åˆ†ææ­¥éª¤</a>
          <ul>
            <li><a href="#æ•°æ®é¢„å¤„ç†">æ•°æ®é¢„å¤„ç†</a></li>
            <li><a href="#åºåˆ—æ¯”å¯¹">åºåˆ—æ¯”å¯¹</a></li>
            <li><a href="#peak-calling">Peak calling</a></li>
            <li><a href="#ç”Ÿæˆbwbigwigæ–‡ä»¶">ç”Ÿæˆbw/bigwigæ–‡ä»¶</a></li>
            <li><a href="#ä¸‹æ¸¸åˆ†æ">ä¸‹æ¸¸åˆ†æ</a>
              <ul>
                <li><a href="#peakæ³¨é‡Š">Peakæ³¨é‡Š</a></li>
                <li><a href="#å³°åŸºå› goå¯Œé›†åˆ†æ">å³°åŸºå› GOå¯Œé›†åˆ†æ</a></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><a href="#å®éªŒç»“æœ">å®éªŒç»“æœ</a>
      <ul>
        <li><a href="#æ•°æ®é¢„å¤„ç†-1">æ•°æ®é¢„å¤„ç†</a></li>
        <li><a href="#åºåˆ—æ¯”å¯¹-1">åºåˆ—æ¯”å¯¹</a></li>
        <li><a href="#peak-calling-1">peak calling</a></li>
        <li><a href="#ä¸‹æ¸¸åˆ†æ-1">ä¸‹æ¸¸åˆ†æ</a></li>
      </ul>
    </li>
    <li><a href="#å®éªŒæ€»ç»“">å®éªŒæ€»ç»“</a></li>
    <li><a href="#å‚è€ƒ">å‚è€ƒ</a></li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h2 id="å®éªŒææ–™">å®éªŒææ–™</h2>
<p><a href="https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&amp;acc=SRR5748809&amp;display=analysis" target="_blank" rel="noopener noreffer ">SRR5748809 : Run Browser : SRA Archive : NCBI (nih.gov)</a></p>
<p>å…³äºç‰ç±³ï¼ˆZea maysï¼‰å¶è‚‰ç»†èƒï¼ˆleaf mesophyllï¼‰çš„ATAC-seqæ•°æ®ï¼Œç”¨äºç ”ç©¶æŸ“è‰²è´¨å¯åŠæ€§çš„é«˜é€šé‡æµ‹åºæ•°æ®</p>
<h2 id="å®éªŒæµç¨‹">å®éªŒæµç¨‹</h2>
<h3 id="è½¯ä»¶">è½¯ä»¶</h3>
<p>SRA-tools <a href="https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/" target="_blank" rel="noopener noreffer ">Index of /sra/sdk (nih.gov)</a></p>
<p>FastQC <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener noreffer ">Babraham Bioinformatics - FastQC A Quality Control tool for High Throughput Sequence Data</a></p>
<p>BWA <a href="https://github.com/lh3/bwa" target="_blank" rel="noopener noreffer ">lh3/bwa: Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment) (github.com)</a></p>
<p>samtools <a href="https://github.com/samtools/samtools" target="_blank" rel="noopener noreffer ">samtools/samtools: Tools (written in C using htslib) for manipulating next-generation sequencing data (github.com)</a></p>
<p>bedtools <a href="https://bedtools.readthedocs.io/en/latest/index.html" target="_blank" rel="noopener noreffer ">bedtools: a powerful toolset for genome arithmetic â€” bedtools 2.31.0 documentation</a></p>
<p>deeptools <a href="https://bedtools.readthedocs.io/en/latest/index.html" target="_blank" rel="noopener noreffer ">bedtools: a powerful toolset for genome arithmetic â€” bedtools 2.31.0 documentation</a></p>
<p>R-clusterPro <a href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank" rel="noopener noreffer ">Bioconductor - clusterProfiler</a></p>
<h3 id="åˆ†ææ­¥éª¤">åˆ†ææ­¥éª¤</h3>
<h4 id="æ•°æ®é¢„å¤„ç†">æ•°æ®é¢„å¤„ç†</h4>
<p>é€šè¿‡itcå¹³å°ä½¿ç”¨é€¸å¤«æ¥¼ç½‘ç»œç¯å¢ƒsshç™»å½•</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">ssh bio_zxdai@172.16.192.8
</span></span></code></pre></td></tr></table>
</div>
</div><p>å°†æ‰€éœ€è¦è½¯ä»¶è·¯å¾„åŠ å…¥ç¯å¢ƒå˜é‡</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">cat &gt;&gt; .bashrc <span class="s">&lt;&lt; EOF
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; PATH=/home/software/sratoolkit.3.1.0-centos_linux64/bin/:$PATH
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; EOF</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>æˆ–è€…å°†ä¸Šè¿°è‹¥å¹²å‘½ä»¤å†™å…¥è„šæœ¬ï¼Œå°†æ‰§è¡Œè„šæœ¬å‘½ä»¤åŠ å…¥ <code>.bashrc</code>æ–‡ä»¶</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 ~<span class="o">]</span>$ cat /home/software/softlist
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/bwa-0.7.17:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/FastQC:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/fastp:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/bowtie2-2.4.5-linux-x86_64:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/hisat2-2.1.0:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/tophat-2.1.1.Linux_x86_64:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/bedtools2/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/samtools-1.9:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/subread-2.0.3-Linux-x86_64/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/stringtie-2.2.1.Linux_x86_64:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/miniconda3/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="c1"># alias samtools=&#34;samtools_0.1.18&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/BatMeth2-master/bin:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nb">alias</span> <span class="nv">Trimmomatic</span><span class="o">=</span><span class="s2">&#34;java -jar /home/software/Trimmomatic-0.39/trimmomatic-0.39.jar&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/STAR-master/source/:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/home/software/qualimap_v2.2.1/:<span class="nv">$PATH</span>
</span></span><span class="line"><span class="cl"><span class="c1">#PATH=/home/software/bowtie-1.3.1/:$PATH</span>
</span></span><span class="line"><span class="cl"><span class="nv">PATH</span><span class="o">=</span>/Bioinfo/zxzhu/software/BatMeth2/bin:<span class="nv">$PATH</span>
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">cat &gt;&gt; .bashrc <span class="s">&lt;&lt; EOF
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; source /home/software/softlist
</span></span></span><span class="line"><span class="cl"><span class="s">&gt; EOF</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>ä½¿ç”¨NCBIçš„sratoolkitä¸‹è½½sraæ•°æ®</p>
<p>prefetchå‘½ä»¤ä¸‹è½½</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">prefetch SRR5748809
</span></span></code></pre></td></tr></table>
</div>
</div><p>fastq-dump å·¥å…·å¯¹ SRA æ–‡ä»¶è¿›è¡Œ fastq æ–‡ä»¶åˆ‡åˆ†</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 data<span class="o">]</span>$ fasterq-dump --split-3 SRR5748809 -O fastq
</span></span><span class="line"><span class="cl">spots <span class="nb">read</span>      : 10,480,210
</span></span><span class="line"><span class="cl">reads <span class="nb">read</span>      : 20,960,420
</span></span><span class="line"><span class="cl">reads written   : 20,960,420
</span></span></code></pre></td></tr></table>
</div>
</div><p>fastqcè´¨é‡æ§åˆ¶</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">fastqc -o ./qc -t <span class="m">4</span> -f fastq ./data/fastq/*
</span></span></code></pre></td></tr></table>
</div>
</div><p>åºåˆ—å‰ªåˆ‡ï¼ˆTrimmomaticï¼‰</p>
<p>åˆ›å»ºTrimmomaticå‘½ä»¤åˆ«å</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="nb">alias</span> <span class="nv">Trimmomatic</span><span class="o">=</span><span class="s2">&#34;java -jar /home/software/Trimmomatic-0.39/trimmomatic-0.39.jar&#34;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>æ·»åŠ adapteræ–‡ä»¶</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">cp /home/software/Trimmomatic-0.39/adapters/* ./
</span></span><span class="line"><span class="cl">cat NexteraPE-PE.fa - TruSeq3-PE-2.fa &gt;&gt; adapters.fa
</span></span></code></pre></td></tr></table>
</div>
</div><p>å•ç«¯æµ‹åºæ•°æ®å‰ªåˆ‡</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">Trimmomatic SE -phred33 -threads <span class="m">3</span> /Bioinfo/bio_2023_2024_2/bio_zxdai/ATAC-seq/data/fastq/SRR5748809.fastq SRR5748809_processed.fastq ILLUMINACLIP:./adapters/adapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:20 MINLEN:36 -trimlog log.txt
</span></span></code></pre></td></tr></table>
</div>
</div><p>å‘½ä»¤å‚æ•°è§£é‡Š</p>
<ul>
<li><code>SE</code>ï¼šè¡¨ç¤ºè¿™æ˜¯å•ç«¯ï¼ˆSingle Endï¼‰æµ‹åºæ•°æ®ã€‚å¦‚æœæ˜¯åŒç«¯ï¼ˆPaired Endï¼‰æµ‹åºæ•°æ®ï¼Œè¿™é‡Œåº”è¯¥ä½¿ç”¨ <code>PE</code>ã€‚</li>
<li><code>-phred33</code>ï¼šæŒ‡å®šè´¨é‡åˆ†æ•°çš„ç¼–ç æ–¹å¼ã€‚Phred33æ˜¯æœ€å¸¸è§çš„ç¼–ç æ–¹å¼ï¼Œç”¨äºIlluminaæ•°æ®å’Œè¾ƒæ–°çš„Sangeræ•°æ®ã€‚</li>
<li><code>-threads 3</code>ï¼šæŒ‡å®šä½¿ç”¨3ä¸ªçº¿ç¨‹è¿›è¡Œå¹¶è¡Œå¤„ç†ã€‚</li>
<li><code>/Bioinfo/bio_2023_2024_2/bio_zxdai/ATAC-seq/data/fastq/SRR5748809.fastq</code>ï¼šè¾“å…¥æ–‡ä»¶çš„è·¯å¾„ï¼Œè¿™æ˜¯éœ€è¦è¿›è¡Œè´¨é‡æ§åˆ¶çš„åŸå§‹æµ‹åºæ•°æ®ã€‚</li>
<li><code>SRR5748809_processed.fastq</code>ï¼šè¾“å‡ºæ–‡ä»¶çš„è·¯å¾„ï¼Œè¿™æ˜¯ç»è¿‡è´¨é‡æ§åˆ¶åçš„æµ‹åºæ•°æ®ã€‚</li>
<li><code>ILLUMINACLIP:./adapters/adapters.fa:2:30:10</code>ï¼šä½¿ç”¨IlluminaClipæ­¥éª¤ç§»é™¤æµ‹åºé€‚é…å™¨ã€‚<code>./adapters/adapters.fa</code>æ˜¯é€‚é…å™¨åºåˆ—çš„æ–‡ä»¶è·¯å¾„ï¼Œ<code>2:30:10</code>æ˜¯è¿™ä¸ªæ­¥éª¤çš„å‚æ•°ï¼Œåˆ†åˆ«è¡¨ç¤ºç§å­åŒ¹é…é•¿åº¦ã€å›æŠ¥å’Œå›æŠ¥å­—ç¬¦ä¸²åŒ¹é…åº¦ã€‚</li>
<li><code>LEADING:3</code>ï¼šç§»é™¤è¯»å–å¼€å§‹éƒ¨åˆ†è´¨é‡ä½äº3çš„ç¢±åŸºã€‚</li>
<li><code>TRAILING:3</code>ï¼šç§»é™¤è¯»å–ç»“æŸéƒ¨åˆ†è´¨é‡ä½äº3çš„ç¢±åŸºã€‚</li>
<li><code>SLIDINGWINDOW:4:20</code>ï¼šä½¿ç”¨æ»‘åŠ¨çª—å£å‰ªåˆ‡ï¼Œå½“çª—å£å†…çš„å¹³å‡è´¨é‡ä½äº20æ—¶å‰ªåˆ‡ã€‚çª—å£å¤§å°ä¸º4ã€‚</li>
<li><code>MINLEN:36</code>ï¼šç§»é™¤é•¿åº¦ä½äº36çš„è¯»å–ã€‚</li>
<li><code>-trimlog log.txt</code>ï¼šè®°å½•æ¯ä¸ªè¯»å–çš„å‰ªåˆ‡ä¿¡æ¯ï¼Œè¾“å‡ºåˆ° <code>log.txt</code>æ–‡ä»¶ã€‚</li>
</ul>
<h4 id="åºåˆ—æ¯”å¯¹">åºåˆ—æ¯”å¯¹</h4>
<p>agstat.txt</p>
<p>å‚è€ƒåŸºå› ç»„ä¸‹è½½</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">wget --quiet  -c https://download.maizegdb.org/Zm-B73-REFERENCE-NAM-5.0/Zm-B73-REFERENCE-NAM-5.0.fa.gz
</span></span><span class="line"><span class="cl"><span class="c1">#è§£å‹å‹ç¼©æ–‡ä»¶</span>
</span></span><span class="line"><span class="cl">gunzip *gz
</span></span></code></pre></td></tr></table>
</div>
</div><p>æ£€æŸ¥åŸºå› ç»„ä¸‹è½½æ˜¯å¦å®Œæ•´</p>
<p>ä½¿ç”¨ <code>md5sum</code>å‘½ä»¤æ¥æ£€æŸ¥æ–‡ä»¶çš„å®Œæ•´æ€§ã€‚è¿™ä¸ªå‘½ä»¤ä¼šè®¡ç®—æ–‡ä»¶çš„MD5å“ˆå¸Œå€¼ï¼Œå¯ä»¥å°†è¿™ä¸ªå€¼ä¸ä½ ä»ä¸‹è½½é¡µé¢å¾—åˆ°çš„MD5å“ˆå¸Œå€¼è¿›è¡Œæ¯”è¾ƒ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">md5sum Zm-B73-REFERENCE-NAM-5.0.fa
</span></span></code></pre></td></tr></table>
</div>
</div><p>BWAå»ºç«‹ç´¢å¼•ã€åŒ¹é…</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#å»ºç«‹ç´¢å¼•</span>
</span></span><span class="line"><span class="cl">nohup bwa index ./ref/Zm*.fa <span class="p">&amp;</span> 
</span></span><span class="line"><span class="cl"><span class="c1">#cdå›ä¸Šä¸€çº§è·¯å¾„</span>
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ..
</span></span></code></pre></td></tr></table>
</div>
</div><p>ä½¿ç”¨ <code>bwa index</code> å»ºç«‹ç´¢å¼•ï¼Œç´¢å¼•æ–‡ä»¶ä¼šä»¥ <code>Zm*.fa</code>æ‰€åŒ¹é…åˆ°çš„è¡¨è¾¾å¼ä¸ºæ–‡ä»¶åå‰ç¼€</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#align</span>
</span></span><span class="line"><span class="cl"><span class="c1">#mem algorithm</span>
</span></span><span class="line"><span class="cl">bwa mem ./data/ref/Zm-B73-REFERENCE-NAM-5.0.fa ./data/fastq/SRR5748809.fastq 1&gt;./align/SRR5748809.sam 2&gt;./align/bwa_mem.log <span class="p">&amp;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#ALN algorithm</span>
</span></span><span class="line"><span class="cl">bwa aln -t <span class="m">5</span> ./data/ref/Zm-B73-REFERENCE-NAM-5.0.fa ./data/fastq/SRR5748809.fastq 1&gt;./align/SRR5748809.sai 2&gt;./align/bwa_aln.log <span class="p">&amp;</span>
</span></span><span class="line"><span class="cl">bwa samse ./data/ref/Zm-B73-REFERENCE-NAM-5.0.fa ./align/SRR5748809.sai ./data/fastq/SRR5748809.fastq 1&gt;./align/SRR5748809.aln.sam <span class="p">&amp;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p><code>bwa mem</code>æ˜¯BWAçš„é»˜è®¤ç®—æ³•ï¼Œé€‚ç”¨äºé•¿åº¦70bp-1Mbpçš„åºåˆ—ã€‚å®ƒæ¯” <code>bwa aln</code>æ›´å¿«ï¼Œæ›´å‡†ç¡®ï¼Œå°¤å…¶æ˜¯å¯¹äºè¾ƒé•¿çš„åºåˆ—ã€‚<code>bwa mem</code>ä¹Ÿæ”¯æŒgapped alignmentï¼ˆå…è®¸æ’å…¥å’Œåˆ é™¤ï¼‰ï¼Œå¹¶ä¸”å¯ä»¥å¤„ç†paired-end reads</p>
<ul>
<li><code>-t å¤šçº¿ç¨‹å¹¶è¡Œ</code></li>
<li><code>1&gt;./align/SRR5748809.sam</code>:  å°†æ ‡å‡†è¾“å‡ºï¼ˆå¯¹é½ç»“æœï¼‰é‡å®šå‘åˆ°SAMæ–‡ä»¶ã€‚</li>
<li><code>2&gt;./align/bwa_mem.log</code>:  å°†æ ‡å‡†é”™è¯¯è¾“å‡ºï¼ˆå¦‚é”™è¯¯å’Œè­¦å‘Šä¿¡æ¯ï¼‰é‡å®šå‘åˆ°æ—¥å¿—æ–‡ä»¶ã€‚</li>
</ul>
<p><code>bwa aln</code>æ˜¯BWAçš„æ—§ç®—æ³•ï¼Œé€‚ç”¨äºé•¿åº¦up to ~100bpçš„åºåˆ—ã€‚å®ƒä½¿ç”¨å›æº¯æ³•ï¼ˆbacktrackingï¼‰è¿›è¡Œæ¯”å¯¹ï¼Œå¯¹äºè¾ƒçŸ­çš„åºåˆ—æ•ˆæœè¾ƒå¥½ã€‚ç„¶è€Œï¼Œ<code>bwa aln</code>ä¸æ”¯æŒgapped alignmentï¼Œå› æ­¤å¯¹äºåŒ…å«æ’å…¥å’Œåˆ é™¤çš„åºåˆ—ï¼Œå…¶æ¯”å¯¹æ•ˆæœå¯èƒ½ä¸å¦‚ <code>bwa mem</code>ã€‚æ¨èä½¿ç”¨ <code>bwa mem</code></p>
<p>alnç®—æ³•ä¼šå…ˆç”Ÿæˆ <code>.sai</code> BWA-backtrackç®—æ³•çš„å¯¹é½ç»“æœæ–‡ä»¶,ç„¶åä½¿ç”¨ <code>bwa samse</code>&ndash;BWAçš„ä¸€ä¸ªæ¨¡å—ï¼Œå°†BWA-backtrackç®—æ³•çš„å¯¹é½ç»“æœè½¬æ¢ä¸ºSAMæ ¼å¼</p>
<p>æ¯”å¯¹ç»“æœçš„ç»Ÿè®¡</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$!</span> &gt; bwa.pid
</span></span><span class="line"><span class="cl"><span class="nb">wait</span> <span class="k">$(</span>cat bwa.pid<span class="k">)</span> <span class="o">&amp;&amp;</span> samtools view -bS ./align/SRR5748809.sam <span class="p">|</span> samtools flagstat - &gt; ./align/flagstat.txt
</span></span></code></pre></td></tr></table>
</div>
</div><p>å°†æµ‹åºæ•°æ®æ¯”å¯¹å›åŸºå› ç»„éœ€è¦èŠ±è´¹ä¸€å®šæ—¶é—´ï¼Œå¯ä»¥ä½¿ç”¨ä¸Šè¿°å‘½ä»¤æˆ–è€…å†™æˆä¸€ä¸ªè„šæœ¬ä»¥è¿è´¯æ‰§è¡Œå‘½ä»¤</p>
<ul>
<li><code>-b</code>: samtoolså°†è¾“å‡ºè½¬æ¢ä¸ºBAMæ ¼å¼</li>
<li><code>-S</code>: æŒ‡å®šsamtoolsè¾“å…¥æ–‡ä»¶SAMæ ¼å¼</li>
<li><code>-</code>: è¡¨ç¤ºä»æ ‡å‡†è¾“å…¥ï¼ˆstdinï¼‰è¯»å–æ•°æ®ï¼Œæ•°æ®æ¥è‡ªäºå‰ä¸€ä¸ªå‘½ä»¤çš„è¾“å‡º</li>
</ul>
<h4 id="peak-calling">Peak calling</h4>
<p>ç­›é€‰æ¯”å¯¹ç»“æœ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="s2">&#34;SRR5748809&#34;</span>
</span></span><span class="line"><span class="cl"><span class="c1">#å»é™¤PCRé‡å¤</span>
</span></span><span class="line"><span class="cl"><span class="nv">$picard</span> MarkDuplicates -I ../align/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.bam 
</span></span><span class="line"><span class="cl">    -O<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.rmdup.bam -M <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.rmdup.metric --REMOVE_DUPLICATES <span class="nb">true</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#å»é™¤ä½è´¨é‡reads(-q 30)ä»¥åŠæœªå¿…å¯¹åˆ°åŒä¸€æ¡æŸ“è‰²ä½“(-f 2)çš„æ•°æ®</span>
</span></span><span class="line"><span class="cl"><span class="nv">$samtools</span> view -h -f <span class="m">2</span> -q <span class="m">30</span> <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.rmdup.bam <span class="p">|</span> grep -v chrM <span class="p">|</span> <span class="nv">$samtools</span> sort -O bam -@ <span class="m">20</span> -o - &gt;<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.last.bam
</span></span><span class="line"><span class="cl">bedtools bamtobed -i <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.last.bam &gt;<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.bed
</span></span></code></pre></td></tr></table>
</div>
</div><p>call peak</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">mkdir peaks 
</span></span><span class="line"><span class="cl">macs2 callpeak -t ./align/*bed -g <span class="m">14300719022</span> --nomodel --shift -100 --extsize <span class="m">200</span> -n SRR5748809 --outdir ./peaks
</span></span></code></pre></td></tr></table>
</div>
</div><p>å‚æ•°çš„è§£æï¼š</p>
<ul>
<li><code>-t</code>ï¼šæŒ‡å®šå¤„ç†ç»„çš„æ–‡ä»¶ï¼Œè¿™é‡Œæ˜¯ BED æ ¼å¼çš„æ–‡ä»¶</li>
<li><code>-g</code>ï¼šæŒ‡å®šåŸºå› ç»„çš„å¤§å°</li>
<li><code>--nomodel</code>ï¼šè¿™ä¸ªé€‰é¡¹å‘Šè¯‰ MACS2 ä¸è¦æ„å»ºç§»ä½æ¨¡å‹ï¼Œè¿™å¯¹äº ATAC-seq æ•°æ®æ˜¯å¿…è¦çš„ï¼Œå› ä¸º ATAC-seq æ•°æ®çš„ç‰‡æ®µå¤§å°ä¸ä»£è¡¨è½¬å½•å› å­ç»“åˆä½ç‚¹ã€‚</li>
<li><code>--shift</code>ï¼šç§»åŠ¨è¯»å–çš„ä¸­å¿ƒä½ç½®ã€‚å¯¹äº ATAC-seq æ•°æ®ï¼Œé€šå¸¸ä¼šè®¾ç½®ä¸º -100ï¼Œè¡¨ç¤ºå‘ 5&rsquo; æ–¹å‘ç§»åŠ¨ 100 bp</li>
<li><code>--extsize</code>ï¼šè®¾ç½®è¯»å–çš„å»¶ä¼¸å¤§å°ã€‚å¯¹äº ATAC-seq æ•°æ®ï¼Œé€šå¸¸ä¼šè®¾ç½®ä¸º 200ï¼Œè¡¨ç¤ºæ¯ä¸ªè¯»å–éƒ½ä¼šå‘ 3&rsquo; æ–¹å‘å»¶ä¼¸ 200 bp</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-text" data-lang="text"><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 tag size is determined as 51 bps 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 tag size = 51.0 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1  total tags in treatment: 27376162 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 user defined the maximum tags... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1  tags after filtering in treatment: 15521232 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1  Redundant rate of treatment: 0.43 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #1 finished! 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Build Peak Model... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Skipped... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Use 200 as fragment length 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #2 Sequencing ends will be shifted towards 5&#39; by 100 bp(s) 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #3 Call peaks... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:32:03: #3 Pre-compute pvalue-qvalue table... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:16: #3 Call peaks for each chromosome... 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:56: #4 Write output xls file... SRR5748809_peaks.xls 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:56: #4 Write peak in narrowPeak format file... SRR5748809_peaks.narrowPeak 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:57: #4 Write summits bed file... SRR5748809_summits.bed 
</span></span><span class="line"><span class="cl">INFO  @ Thu, 28 Mar 2024 15:33:57: Done! 
</span></span></code></pre></td></tr></table>
</div>
</div><p>ä¸ä½¿ç”¨nomodelå‚æ•°å¯ä»¥è·å¾—model_plot.rï¼Œç”¨äºç»˜åˆ¶ç‰‡æ®µå¤§å°çš„åˆ†å¸ƒå›¾ï¼Œåæ˜ äº†å¼€æ”¾æŸ“è‰²è´¨çš„åŒºåŸŸ</p>
<h4 id="ç”Ÿæˆbwbigwigæ–‡ä»¶">ç”Ÿæˆbw/bigwigæ–‡ä»¶</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712560424603.png"
        title="1712560424603" width="551" height="435" /></p>
<p>é€šè¿‡è·¯å¾„ bam -&gt; bedGraph -&gt; bigwig/bw</p>
<p>è·å– <code>chrom.sizes</code> æ–‡ä»¶ï¼Œå…¶ä¸­åŒ…å«äº†æ¯ä¸ªæŸ“è‰²ä½“çš„åç§°å’Œå¤§å°</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">samtools faidx data/ref/*fa
</span></span><span class="line"><span class="cl">cut -f1,2 data/ref/*.fa.fai &gt; data/ref/chrom.sizes
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">bedtools genomecov -bg -i peaks/*summits.bed -g data/ref/chrom.sizes &gt; peaks/SRR5748809.bedGraph
</span></span><span class="line"><span class="cl">bedGraphToBigWig peaks/SRR5748809.bedGraph data/ref/chrom.sizes peaks/SRR5748809.bw
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 ATAC-seq<span class="o">]</span>$ ll peaks
</span></span><span class="line"><span class="cl">total <span class="m">12524</span>
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">1288417</span> Apr  <span class="m">8</span> 15:22 SRR5748809.bedGraph
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai  <span class="m">559174</span> Apr  <span class="m">8</span> 15:22 SRR5748809.bw
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai   <span class="m">25600</span> Apr  <span class="m">8</span> 14:33 SRR5748809_model.pdf
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai   <span class="m">99228</span> Mar <span class="m">28</span> 15:47 SRR5748809_model.r
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">3919519</span> Mar <span class="m">28</span> 15:49 SRR5748809_peaks.narrowPeak
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">4257998</span> Mar <span class="m">28</span> 15:49 SRR5748809_peaks.xls
</span></span><span class="line"><span class="cl">-rw-rw-r--. <span class="m">1</span> bio_zxdai bio_zxdai <span class="m">2660699</span> Mar <span class="m">28</span> 15:49 SRR5748809_summits.bed
</span></span></code></pre></td></tr></table>
</div>
</div><p>ä¸Šæ¸¸åˆ†æç»“æŸ</p>
<h4 id="ä¸‹æ¸¸åˆ†æ">ä¸‹æ¸¸åˆ†æ</h4>
<h5 id="peakæ³¨é‡Š">Peakæ³¨é‡Š</h5>
<p>åªè€ƒè™‘ç‰ç±³æŸ“è‰²ä½“1-10</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span><span class="lnt">42
</span><span class="lnt">43
</span><span class="lnt">44
</span><span class="lnt">45
</span><span class="lnt">46
</span><span class="lnt">47
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-R" data-lang="R"><span class="line"><span class="cl"><span class="c1">#BiocManager::install(&#34;GenomicFeatures&#34;)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#BiocManager::install(&#34;ChIPseeker&#34;)</span>
</span></span><span class="line"><span class="cl"><span class="nf">install.packages</span><span class="p">(</span><span class="s">&#34;ggupset&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">ChIPseeker</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#install.packages(&#34;RMariaDB&#34;)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">RMariaDB</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">GenomicFeatures</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">ggimage</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">maize_TxDb</span> <span class="o">&lt;-</span> <span class="nf">makeTxDbFromGFF</span><span class="p">(</span><span class="s">&#34;D:/seq_project/seq_project/work_data/Zma_B73_V4.gff3&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">wd</span> <span class="o">&lt;-</span> <span class="s">&#34;F:/course/ç”Ÿç‰©ä¿¡æ¯å­¦è¿›å±•/ATAC-seq/ATAC&#34;</span>
</span></span><span class="line"><span class="cl"><span class="nf">setwd</span><span class="p">(</span><span class="n">wd</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">tissue</span> <span class="o">&lt;-</span> <span class="s">&#34;Zea_mays&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">bed_file</span> <span class="o">&lt;-</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="s">&#34;SRR5748809_summits.bed&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">bed_name</span> <span class="o">&lt;-</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;_summits.*&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">))</span>
</span></span><span class="line"><span class="cl"><span class="nf">names</span><span class="p">(</span><span class="n">bed_file</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">bed_name</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">bed</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="n">bed_file</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">bed</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">bed</span><span class="p">,</span> <span class="n">bed</span><span class="o">$</span><span class="n">V1</span> <span class="o">%in%</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;chr&#34;</span><span class="p">,</span> <span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">))</span> <span class="o">%&gt;%</span>
</span></span><span class="line"><span class="cl">        <span class="nf">mutate</span><span class="p">(</span><span class="n">V1</span> <span class="o">=</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;chr&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">V1</span><span class="p">))</span> 
</span></span><span class="line"><span class="cl"><span class="nf">write.table</span><span class="p">(</span><span class="n">bed</span><span class="p">,</span> <span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">bed_name</span><span class="p">,</span> <span class="s">&#34;.bed&#34;</span><span class="p">)),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">bed_file</span> <span class="o">&lt;-</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">bed_name</span><span class="p">,</span> <span class="s">&#34;.bed&#34;</span><span class="p">))</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">peak</span> <span class="o">&lt;-</span> <span class="nf">readPeakFile</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">bed_name</span><span class="p">,</span> <span class="s">&#34;.bed&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="c1"># # å°†æŸ“è‰²ä½“åç§°çš„æ ¼å¼æ”¹ä¸º &#34;chr&#34; å¼€å¤´</span>
</span></span><span class="line"><span class="cl"><span class="c1"># seqlevelsStyle(maize_TxDb) &lt;- &#34;UCSC&#34;</span>
</span></span><span class="line"><span class="cl"><span class="c1"># # æ£€æŸ¥æ”¹å˜åçš„æŸ“è‰²ä½“åç§°</span>
</span></span><span class="line"><span class="cl"><span class="c1"># seqlevels(maize_TxDb)s</span>
</span></span><span class="line"><span class="cl"><span class="n">peakAnno</span> <span class="o">&lt;-</span> <span class="nf">annotatePeak</span><span class="p">(</span><span class="n">peak</span><span class="p">,</span> <span class="n">tssRegion</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span> <span class="m">1000</span><span class="p">),</span> <span class="n">TxDb</span><span class="o">=</span><span class="n">maize_TxDb</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_pie&#34;</span><span class="p">,</span> <span class="s">&#34;.pdf&#34;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="nf">plotAnnoPie</span><span class="p">(</span><span class="n">peakAnno</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_upset&#34;</span><span class="p">,</span> <span class="s">&#34;.pdf&#34;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">upsetplot</span><span class="p">(</span><span class="n">peakAnno</span><span class="p">,</span> <span class="n">vennpie</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="s">&#34;genebody_PeakProf&#34;</span><span class="p">,</span> <span class="s">&#34;.pdf&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="n">p_1</span> <span class="o">&lt;-</span> <span class="nf">plotPeakProf2</span><span class="p">(</span><span class="n">bed_file</span><span class="p">,</span> <span class="n">upstream</span> <span class="o">=</span> <span class="m">1000</span><span class="p">,</span> <span class="n">downstream</span> <span class="o">=</span> <span class="m">1000</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">        <span class="n">conf</span> <span class="o">=</span> <span class="m">0.95</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="s">&#34;gene&#34;</span><span class="p">,</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&#34;body&#34;</span><span class="p">,</span> <span class="n">nbin</span> <span class="o">=</span> <span class="m">800</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">        <span class="n">TxDb</span> <span class="o">=</span> <span class="n">maize_TxDb</span><span class="p">,</span> <span class="n">facet</span> <span class="o">=</span> <span class="s">&#34;row&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">p_1</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>TSS/gene body ä½ç‚¹peak plotå±•ç¤º</p>
<p>deeptools</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">computeMatrix scale-regions -p <span class="m">12</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -S ./peaks/SRR5748809.bw <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -R ./peaks/SRR5748809_summits.bed <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -b <span class="m">100000</span> -a <span class="m">100000</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o ./plot/SRR5748809.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --binSize <span class="m">10</span> --missingDataAsZero <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --skipZeros <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --outFileNameMatrix ./plot/SRR5748809_matrix.tab
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">plotHeatmap -m ./plot/SRR5748809.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -out ./plot/SRR5748809.pdf <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --dpi <span class="m">300</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --colorMap YlGnBu  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --missingDataColor <span class="s2">&#34;#FFF6EB&#34;</span><span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --heatmapHeight <span class="m">21</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --startLabel <span class="s2">&#34;TSS&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --endLabel <span class="s2">&#34;TTS&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --regionsLabel <span class="s2">&#34;gene&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --legendLocation <span class="s2">&#34;none&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --plotTitle <span class="s2">&#34;Zea mays&#34;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><img src="image/index/1713284604918.png" alt="ATAC-seqå³°å€¼å›¾&çƒ­å›¾" height=800>
<h5 id="å³°åŸºå› goå¯Œé›†åˆ†æ">å³°åŸºå› GOå¯Œé›†åˆ†æ</h5>
<p>Gene Ontology</p>
<p>ä½¿ç”¨ <code>ChIPseeker</code>åŒ…å°†å³°å€¼æ³¨é‡Šåˆ°æœ€è¿‘çš„åŸºå› ï¼ŒGOå¯Œé›†åˆ†æ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-R" data-lang="R"><span class="line"><span class="cl"><span class="n">peakAnno</span> <span class="o">&lt;-</span> <span class="nf">annotatePeak</span><span class="p">(</span><span class="n">peak</span><span class="p">,</span> <span class="n">tssRegion</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span> <span class="m">1000</span><span class="p">),</span> <span class="n">TxDb</span><span class="o">=</span><span class="n">maize_TxDb</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#annotated peaks to gene and clusterProfiler</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">AnnotationHub</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">hub</span> <span class="o">&lt;-</span> <span class="nf">AnnotationHub</span><span class="p">()</span>
</span></span><span class="line"><span class="cl"><span class="n">maize</span> <span class="o">&lt;-</span> <span class="n">hub[[</span><span class="s">&#34;AH114308&#34;</span><span class="n">]]</span>
</span></span><span class="line"><span class="cl"><span class="n">entrez_gene_id</span> <span class="o">&lt;-</span> <span class="nf">read.csv</span><span class="p">(</span><span class="s">&#34;D:/motif_project/16.tss_maxmotif_GO/data/entrez_Zm_gene_v4.csv&#34;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="n">gene_ids</span> <span class="o">&lt;-</span> <span class="nf">unique</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">peakAnno</span><span class="p">)</span><span class="o">$</span><span class="n">geneId</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">as.data.frame</span><span class="p">()</span>
</span></span><span class="line"><span class="cl"><span class="nf">colnames</span><span class="p">(</span><span class="n">gene_ids</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="s">&#34;geneId&#34;</span>
</span></span><span class="line"><span class="cl"><span class="n">genes</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">gene_ids</span><span class="p">,</span> <span class="n">entrez_gene_id</span><span class="p">,</span> <span class="n">by.x</span> <span class="o">=</span> <span class="s">&#34;geneId&#34;</span><span class="p">,</span> <span class="n">by.y</span> <span class="o">=</span> <span class="s">&#34;gene_name&#34;</span><span class="p">,</span> <span class="n">all.x</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> <span class="o">%&gt;%</span>
</span></span><span class="line"><span class="cl">        <span class="nf">na.omit</span><span class="p">()</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="n">ego</span> <span class="o">&lt;-</span> <span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span>         <span class="o">=</span> <span class="n">genes</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">OrgDb</span>        <span class="o">=</span> <span class="n">maize</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">keyType</span>      <span class="o">=</span> <span class="s">&#34;ENTREZID&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">pAdjustMethod</span> <span class="o">=</span> <span class="s">&#34;BH&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">pvalueCutoff</span> <span class="o">=</span> <span class="m">0.8</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">                <span class="n">qvalueCutoff</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span>
</span></span><span class="line"><span class="cl">                <span class="n">readable</span>     <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span> 
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">head</span><span class="p">(</span><span class="n">ego</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">write.table</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span> <span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_GO.tsv&#34;</span><span class="p">)),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">quote</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">file.path</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span> <span class="nf">paste0</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">bed_file</span><span class="p">),</span> <span class="s">&#34;_GO_barplot.pdf&#34;</span><span class="p">)))</span>
</span></span><span class="line"><span class="cl"><span class="n">dotp</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span> 
</span></span><span class="line"><span class="cl">        <span class="nf">aes</span><span class="p">(</span><span class="n">GeneRatio</span><span class="p">,</span> <span class="nf">fct_reorder</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span> <span class="n">GeneRatio</span><span class="p">)))</span> <span class="o">+</span> 
</span></span><span class="line"><span class="cl">        <span class="nf">geom_segment</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">xend</span><span class="o">=</span><span class="m">0</span><span class="p">,</span> <span class="n">yend</span> <span class="o">=</span> <span class="n">Description</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">p.adjust</span><span class="p">,</span> <span class="n">size</span> <span class="o">=</span> <span class="n">Count</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">scale_color_viridis_c</span><span class="p">(</span><span class="n">guide</span><span class="o">=</span><span class="nf">guide_colorbar</span><span class="p">(</span><span class="n">reverse</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">scale_size_continuous</span><span class="p">(</span><span class="n">range</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span> <span class="m">10</span><span class="p">))</span> <span class="o">+</span>
</span></span><span class="line"><span class="cl">        <span class="nf">theme_minimal</span><span class="p">()</span> <span class="o">+</span> 
</span></span><span class="line"><span class="cl">        <span class="nf">ylab</span><span class="p">(</span><span class="kc">NULL</span><span class="p">)</span>
</span></span><span class="line"><span class="cl"><span class="nf">print</span><span class="p">(</span><span class="n">dotp</span><span class="p">)</span> 
</span></span><span class="line"><span class="cl"><span class="nf">dev.off</span><span class="p">()</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="å®éªŒç»“æœ">å®éªŒç»“æœ</h2>
<h3 id="æ•°æ®é¢„å¤„ç†-1">æ•°æ®é¢„å¤„ç†</h3>
<p>åŸå§‹SRR5748809 QC</p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png" title="1710343095852" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png" data-sub-html="<h2>åŸå§‹æ•°æ®FastQC summary</h2><p>1710343095852</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343095852.png" width="470" height="699" />
    </a><figcaption class="image-caption">åŸå§‹æ•°æ®FastQC summary</figcaption>
    </figure></p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png" title="1710343119909" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png" data-sub-html="<h2>åŸå§‹æ•°æ® Bsic Statistics</h2><p>1710343119909</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343119909.png" width="1072" height="404" />
    </a><figcaption class="image-caption">åŸå§‹æ•°æ® Bsic Statistics</figcaption>
    </figure></p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png" title="1710343138678" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png" data-sub-html="<h2>åŸå§‹æ•°æ®è´¨é‡ç®±çº¿å›¾åˆ†å¸ƒ</h2><p>1710343138678</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343138678.png" width="1175" height="806" />
    </a><figcaption class="image-caption">åŸå§‹æ•°æ®è´¨é‡ç®±çº¿å›¾åˆ†å¸ƒ</figcaption>
    </figure></p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png" title="1710343385005" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png" data-sub-html="<h2>åŸå§‹æ•°æ®æ— è¿‡è¡¨è¾¾åºåˆ—å’Œæ¥å¤´</h2><p>1710343385005</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343385005.png" width="1243" height="999" />
    </a><figcaption class="image-caption">åŸå§‹æ•°æ®æ— è¿‡è¡¨è¾¾åºåˆ—å’Œæ¥å¤´</figcaption>
    </figure></p>
<p>ä½¿ç”¨ä¸Šè¿°Trimmomaticå‘½ä»¤å‰ªåˆ‡åï¼Œè´¨é‡æ²¡æœ‰ä¹‹å‰å¥½</p>
<p><figure><a class="lightgallery" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png" title="1710343706916" data-thumbnail="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png" data-sub-html="<h2>å‰ªåˆ‡æ•°æ®è´¨é‡summary</h2><p>1710343706916</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png"
            data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png 2x"
            data-sizes="auto"
            alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343706916.png" width="344" height="486" />
    </a><figcaption class="image-caption">å‰ªåˆ‡æ•°æ®è´¨é‡summary</figcaption>
    </figure></p>
<p>å…¶ä¸­per base sequence contentæœ‰äº›è®¸æ”¹å˜ï¼Œä¼¼ä¹å¯ä»¥é€šè¿‡ <code>HEADCROP:14</code>å‚æ•°åˆ é™¤å‰14ä¸ªç¢±åŸºï¼Œä½†åºåˆ—æœ¬èº«ä¸æ˜¯å¾ˆé•¿ï¼Œä¾¿æ²¡æœ‰å°è¯•</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1710343744796.png"
        title="1710343744796" width="1119" height="796" /></p>
<h3 id="åºåˆ—æ¯”å¯¹-1">åºåˆ—æ¯”å¯¹</h3>
<p><a href="/2024_spring_ATAC/flagstat.txt" rel="">åºåˆ—æ¯”å¯¹æ•°æ®ç»“æœ</a></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">[</span>bio_zxdai@bioinfo-01 align<span class="o">]</span>$ cat flagstat.txt
</span></span><span class="line"><span class="cl"><span class="m">27933221</span> + <span class="m">0</span> in total <span class="o">(</span>QC-passed reads + QC-failed reads<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> secondary
</span></span><span class="line"><span class="cl"><span class="m">34</span> + <span class="m">0</span> supplementary
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> duplicates
</span></span><span class="line"><span class="cl"><span class="m">27376162</span> + <span class="m">0</span> mapped <span class="o">(</span>98.01% : N/A<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> paired in sequencing
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> read1
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> read2
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> properly paired <span class="o">(</span>N/A : N/A<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> with itself and mate mapped
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> singletons <span class="o">(</span>N/A : N/A<span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> with mate mapped to a different chr
</span></span><span class="line"><span class="cl"><span class="m">0</span> + <span class="m">0</span> with mate mapped to a different chr <span class="o">(</span>mapQ&gt;<span class="o">=</span>5<span class="o">)</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="peak-calling-1">peak calling</h3>
<p>peak callingç»“æœåˆ†æ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 tag size is determined as 51 bps </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 tag size = 51.0 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1  total tags in treatment: 27376162 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 user defined the maximum tags... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1  tags after filtering in treatment: 15521232 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1  Redundant rate of treatment: 0.43 </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#1 finished! </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Build Peak Model... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Skipped... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Use 200 as fragment length </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#2 Sequencing ends will be shifted towards 5&#39; by 100 bp(s) </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#3 Call peaks... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:32:03: <span class="c1">#3 Pre-compute pvalue-qvalue table... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:16: <span class="c1">#3 Call peaks for each chromosome... </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:56: <span class="c1">#4 Write output xls file... SRR5748809_peaks.xls </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:56: <span class="c1">#4 Write peak in narrowPeak format file... SRR5748809_peaks.narrowPeak </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:57: <span class="c1">#4 Write summits bed file... SRR5748809_summits.bed </span>
</span></span><span class="line"><span class="cl">INFO  @ Thu, <span class="m">28</span> Mar <span class="m">2024</span> 15:33:57: Done! 
</span></span></code></pre></td></tr></table>
</div>
</div><ul>
<li><code>tag size is determined as 51 bps</code>ï¼šreadsçš„å¤§å°ä¸º51ä¸ªç¢±åŸºå¯¹</li>
<li><code>filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s)</code>ï¼šåŒä¸€ä½ç½®å’ŒåŒä¸€é“¾ä¸Šçš„å†—ä½™æ ‡ç­¾è¢«è¿‡æ»¤æ‰</li>
<li><code>tags after filtering in treatment: 15521232</code>ï¼šè¿‡æ»¤åï¼Œå¤„ç†çš„æ ·æœ¬ä¸­æœ‰15521232ä¸ªtags</li>
<li><code>Redundant rate of treatment: 0.43</code>ï¼šå¤„ç†çš„æ ·æœ¬ä¸­çš„å†—ä½™ç‡ä¸º0.43</li>
</ul>
<p><a href="/2024_spring_ATAC/SRR5748809_model.pdf" rel="">SRR5748809_model.pdf</a></p>
<p>åæ˜ äº†å¼€æ”¾æŸ“è‰²è´¨çš„åŒºåŸŸï¼Œè¿™ä¸ªå›¾æ˜¾ç¤ºå‡ºå¤šä¸ªå³°å€¼ï¼Œè¿™äº›å³°å€¼å¯¹åº”äºä¸åŒçš„æ ¸å°ä½“çŠ¶æ€</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559028672.png"
        title="1712559028672" width="796" height="698" /></p>
<p>äº¤å‰ç›¸å…³æ€§ï¼ˆCross-correlationï¼‰å›¾ï¼Œxè½´ä»£è¡¨ç€æ»åï¼ˆlagï¼‰ï¼Œä¹Ÿå°±æ˜¯è¯»å–çš„ç§»ä½é‡ï¼Œyè½´ä»£è¡¨ç€äº¤å‰ç›¸å…³æ€§ï¼Œä¹Ÿå°±æ˜¯åœ¨ç»™å®šçš„æ»åä¸‹ï¼Œæ­£å‘å’Œåå‘è¯»å–çš„ç›¸å…³æ€§ï¼Œå³°å€¼çš„ä½ç½®è¡¨ç¤ºäº†æœ€ä½³çš„æ»åï¼Œä¹Ÿå°±æ˜¯æœ€ä½³çš„ç‰‡æ®µå¤§å°</p>
<p>åªæœ‰ä¸€ä¸ªæ˜æ˜¾çš„å³°å€¼ï¼Œè¿™è¡¨ç¤ºå¤§éƒ¨åˆ†çš„è¯»å–éƒ½é›†ä¸­åœ¨ç»“åˆä½ç‚¹é™„è¿‘</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712559047974.png"
        title="1712559047974" width="793" height="787" /></p>
<h3 id="ä¸‹æ¸¸åˆ†æ-1">ä¸‹æ¸¸åˆ†æ</h3>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728314927.png"
        title="1712728314927" width="999" height="308" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1712728353944.png"
        title="1712728353944" width="835" height="832" /></p>
<p>GOå¯Œé›†åˆ†æ</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png"
        data-srcset="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png 1.5x, /%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png 2x"
        data-sizes="auto"
        alt="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/image/index/1713284842478.png"
        title="1713284842478" width="740" height="654" /></p>
<h2 id="å®éªŒæ€»ç»“">å®éªŒæ€»ç»“</h2>
<p>åŸå§‹æµ‹åºæ•°æ®ä¸é”™ï¼Œä½¿ç”¨FastQCè´¨é‡æ§åˆ¶æŸ¥çœ‹ä¸€ä¸‹ï¼Œä¿è¯</p>
<ol>
<li>åºåˆ—è´¨é‡åˆæ ¼</li>
<li>æ²¡æœ‰è¿‡è¡¨è¾¾çš„primer adaptor</li>
<li>åºåˆ—é•¿åº¦ä¸æ˜¯å¾ˆçŸ­ å³å¯</li>
</ol>
<p>åºåˆ—æ¯”å¯¹æ•°æ®ç»“æœåˆ†æ</p>
<ul>
<li><code>27933221 + 0 in total (QC-passed reads + QC-failed reads)</code>: æ€»å…±æœ‰27933221ä¸ªreadsï¼Œå…¶ä¸­æ²¡æœ‰QCå¤±è´¥çš„readsã€‚æµ‹åºæ•°æ®çš„è´¨é‡åº”è¯¥æ˜¯ç›¸å½“å¥½çš„</li>
<li><code>0 + 0 secondary</code>: æ²¡æœ‰secondaryï¼ˆæ¬¡è¦çš„ï¼‰å¯¹é½ï¼Œè¿™äº›é€šå¸¸æ˜¯ç”±äºreadså¯ä»¥å¯¹é½åˆ°å¤šä¸ªä½ç½®ã€‚</li>
<li><code>34 + 0 supplementary</code>: æœ‰34ä¸ªè¡¥å……çš„å¯¹é½ï¼Œè¿™äº›é€šå¸¸æ˜¯ç”±äºreadså¯ä»¥å¯¹é½åˆ°å¤šä¸ªä½ç½®ã€‚</li>
<li><code>27376162 + 0 mapped (98.01% : N/A)</code>: æœ‰27376162ä¸ªreadsè¢«æ˜ å°„åˆ°å‚è€ƒåºåˆ—ï¼Œå æ€»readsçš„98.01%ã€‚éå¸¸é«˜çš„æ˜ å°„ç‡ï¼Œè¯´æ˜å¤§éƒ¨åˆ†readséƒ½èƒ½æ‰¾åˆ°å‚è€ƒåºåˆ—ä¸Šçš„å¯¹åº”ä½ç½®</li>
<li><code>0 + 0 paired in sequencing</code>ã€<code>0 + 0 read1</code>ã€<code>0 + 0 read2</code>ã€<code>0 + 0 properly paired (N/A : N/A)</code>ã€<code>0 + 0 with itself and mate mapped</code>ã€<code>0 + 0 singletons (N/A : N/A)</code>ã€<code>0 + 0 with mate mapped to a different chr</code>ã€<code>0 + 0 with mate mapped to a different chr (mapQ&gt;=5)</code>: è¿™äº›ç»Ÿè®¡ä¿¡æ¯éƒ½æ˜¯å…³äºé…å¯¹çš„readsçš„ï¼Œä½†åœ¨è¿™ä¸ªæ•°æ®é›†ä¸­ï¼Œæ‰€æœ‰çš„readséƒ½æ˜¯å•ç«¯çš„ï¼Œæ‰€ä»¥è¿™äº›ç»Ÿè®¡ä¿¡æ¯éƒ½æ˜¯0</li>
</ul>
<p>åŸºå› å¯Œé›†åˆ†æç»“è®º</p>
<p>GOå¯Œé›†åˆ†æç»“æœæ˜¾ç¤ºäº†&quot;protein-containing complex binding&quot;ã€&ldquo;cytoskeletal motor activity&quot;å’Œ&quot;microtubule motor activity&quot;è¿™ä¸‰ä¸ªGOé¡¹ã€‚è¿™å¯èƒ½æ„å‘³ç€åœ¨å¶è‚‰ç»†èƒä¸­ï¼Œæœ‰ä¸€äº›åŸºå› çš„è°ƒæ§åŒºåŸŸæ˜¯å¼€æ”¾çš„ï¼Œè¿™äº›åŸºå› å¯èƒ½å‚ä¸åˆ°è›‹ç™½è´¨å¤åˆç‰©çš„ç»“åˆå’Œç»†èƒéª¨æ¶çš„è¿åŠ¨ä¸­ã€‚è¿™äº›è¿‡ç¨‹å¯èƒ½ä¸å¶è‚‰ç»†èƒçš„ä¸€äº›é‡è¦åŠŸèƒ½æœ‰å…³ï¼Œä¾‹å¦‚å…‰åˆä½œç”¨ã€ç‰©è´¨è¿è¾“ç­‰</p>
<ol>
<li>&ldquo;protein-containing complex binding&rdquo;ï¼šæ¶‰åŠåˆ°çš„åŸºå› ç¼–ç çš„è›‹ç™½è´¨èƒ½å¤Ÿä¸åŒ…å«å¤šç§è›‹ç™½è´¨çš„å¤åˆç‰©ç»“åˆã€‚è¿™å¯èƒ½æ„å‘³ç€ATACå³°å€¼é™„è¿‘åŸºå› å¯èƒ½å‚ä¸åˆ°ä¸€äº›éœ€è¦å¤šç§è›‹ç™½è´¨å…±åŒä½œç”¨çš„ç”Ÿç‰©è¿‡ç¨‹ä¸­ï¼Œä¾‹å¦‚ä¿¡å·è½¬å¯¼ã€åŸºå› è½¬å½•è°ƒæ§ç­‰ã€‚</li>
<li>&ldquo;cytoskeletal motor activity&quot;å’Œ&quot;microtubule motor activity&rdquo;ï¼šè¿™ä¸¤ä¸ªGOé¡¹éƒ½ä¸ç»†èƒéª¨æ¶å’Œå¾®ç®¡è¿åŠ¨æœ‰å…³ï¼Œå‚ä¸åˆ°è®¸å¤šé‡è¦çš„ç»†èƒè¿‡ç¨‹ä¸­ï¼Œä¾‹å¦‚ç»†èƒåˆ†è£‚ã€ç‰©è´¨è¿è¾“ç­‰ã€‚</li>
</ol>
<h2 id="å‚è€ƒ">å‚è€ƒ</h2>
<p><a href="https://www.cnblogs.com/djx571/p/9493934.html" target="_blank" rel="noopener noreffer ">fasta/fastqæ ¼å¼è§£è¯» - å‘é‚£ä¸ªå¤ªä¸¢äºº - åšå®¢å›­ (cnblogs.com)</a></p>
<p><a href="https://cloud.tencent.com/developer/article/1772411" target="_blank" rel="noopener noreffer ">ç”Ÿä¿¡è½¯ä»¶ | Sratools (æ“ä½œSRAæ–‡ä»¶)-è…¾è®¯äº‘å¼€å‘è€…ç¤¾åŒº-è…¾è®¯äº‘ (tencent.com)</a></p>
<p><a href="https://www.jianshu.com/p/bdfa8f7e5a61" target="_blank" rel="noopener noreffer ">è½¬å½•ç»„åˆ†æâ€”â€”ä¸‰ã€sraè½¬æ¢æˆfastq - ç®€ä¹¦ (jianshu.com)</a></p>
<p><a href="https://www.jianshu.com/p/5c97a34cc1ad" target="_blank" rel="noopener noreffer ">éƒ½8102å¹´äº†ï¼Œè¿˜ç”¨fastq-dumpï¼Œå¿«æ¢fasterq-dumpå§ - ç®€ä¹¦ (jianshu.com)</a></p>
<p><a href="https://zhuanlan.zhihu.com/p/642000061" target="_blank" rel="noopener noreffer ">è½¬å½•ç»„ä¹‹åºåˆ—å‰ªåˆ‡ï¼ˆTrimmomaticï¼‰[å­¦ä¹ ç¬”è®°é€šä¿—æ˜“æ‡‚ç‰ˆ] - çŸ¥ä¹ (zhihu.com)</a></p>
<p><a href="https://zhuanlan.zhihu.com/p/415718382" target="_blank" rel="noopener noreffer ">ã€Œä¸å›½åŒåº†ï¼Œä¸‡å­—é•¿æ–‡ã€ATAC-seqå®æˆ˜æ•™ç¨‹ï¼šä»SRAæ•°æ®ä¸‹è½½åˆ°é«˜åˆ†è¾¨ç‡è®ºæ–‡ä¸»å›¾ç»˜åˆ¶ - çŸ¥ä¹ (zhihu.com)</a></p>
</div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>æ›´æ–°äº 2024-03-09</span>
            </div></div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/index.md" target="_blank">é˜…è¯»åŸå§‹æ–‡æ¡£</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="åˆ†äº«åˆ° Twitter" data-sharer="twitter" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†" data-via="xxxx"><i class="fab fa-twitter fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="åˆ†äº«åˆ° Facebook" data-sharer="facebook" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/"><i class="fab fa-facebook-square fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="åˆ†äº«åˆ° Hacker News" data-sharer="hackernews" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†"><i class="fab fa-hacker-news fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="åˆ†äº«åˆ° Line" data-sharer="line" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†"><i data-svg-src="/lib/simple-icons/icons/line.min.svg" aria-hidden="true"></i></a><a href="javascript:void(0);" title="åˆ†äº«åˆ° å¾®åš" data-sharer="weibo" data-url="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" data-title="ç”Ÿä¿¡è¿›å±•-ATACæ•°æ®å¤„ç†"><i class="fab fa-weibo fa-fw" aria-hidden="true"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">è¿”å›</a></span>&nbsp;|&nbsp;<span><a href="/">ä¸»é¡µ</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/molecular-biology/" class="prev" rel="prev" title="ç°ä»£åˆ†å­ç”Ÿç‰©"><i class="fas fa-angle-left fa-fw" aria-hidden="true"></i>ç°ä»£åˆ†å­ç”Ÿç‰©</a>
            <a href="/ngs_visualization/" class="next" rel="next" title="NGSå¯è§†åŒ–">NGSå¯è§†åŒ–<i class="fas fa-angle-right fa-fw" aria-hidden="true"></i></a></div>
</div>
<div id="comments"></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">ç”± <a href="https://gohugo.io/" target="_blank" rel="noopener noreffer" title="Hugo 0.125.4">Hugo</a> å¼ºåŠ›é©±åŠ¨ | ä¸»é¢˜ - <a href="https://github.com/dillonzq/LoveIt" target="_blank" rel="noopener noreffer" title="LoveIt 0.2.11"><i class="far fa-kiss-wink-heart fa-fw" aria-hidden="true"></i> LoveIt</a>
                </div><div class="footer-line" itemscope itemtype="http://schema.org/CreativeWork"><i class="far fa-copyright fa-fw" aria-hidden="true"></i><span itemprop="copyrightYear">2023 - 2024</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="http://mail.qq.com/cgi-bin/qm_share?t=qm_mailme&amp;email=lOf88frw4aSip6TU5eW69-v5" target="_blank">griedzx</a></span>&nbsp;|&nbsp;<span class="license"><a rel="license external nofollow noopener noreffer" href="https://creativecommons.org/licenses/by-nc/4.0/" target="_blank">CC BY-NC 4.0</a></span></div>
        </div>

        
        <div class="busuanzi-footer">
        <span id="busuanzi_container_site_pv">
            æœ¬ç«™æ€»è®¿é—®é‡<span id="busuanzi_value_site_pv"></span>æ¬¡
        </span>
        <span id="busuanzi_container_site_uv">
            æœ¬ç«™è®¿å®¢æ•°<span id="busuanzi_value_site_uv"></span>äººæ¬¡
        </span>
        </div></footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="å›åˆ°é¡¶éƒ¨">
                <i class="fas fa-arrow-up fa-fw" aria-hidden="true"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="æŸ¥çœ‹è¯„è®º">
                <i class="fas fa-comment fa-fw" aria-hidden="true"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/cookieconsent/cookieconsent.min.css"><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.stemmer.support.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.zh.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/mhchem.min.js"></script><script type="text/javascript" src="/lib/cookieconsent/cookieconsent.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"å¤åˆ¶åˆ°å‰ªè´´æ¿","maxShownLines":50},"comment":{},"cookieconsent":{"content":{"dismiss":"åŒæ„","link":"äº†è§£æ›´å¤š","message":"æœ¬ç½‘ç«™ä½¿ç”¨ Cookies æ¥æ”¹å–„æ‚¨çš„æµè§ˆä½“éªŒ."},"enable":true,"palette":{"button":{"background":"#f0f0f0"},"popup":{"background":"#1aa3ff"}},"theme":"edgeless"},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":true,"left":"\\begin{equation}","right":"\\end{equation}"},{"display":true,"left":"\\begin{equation*}","right":"\\end{equation*}"},{"display":true,"left":"\\begin{align}","right":"\\end{align}"},{"display":true,"left":"\\begin{align*}","right":"\\end{align*}"},{"display":true,"left":"\\begin{alignat}","right":"\\end{alignat}"},{"display":true,"left":"\\begin{alignat*}","right":"\\end{alignat*}"},{"display":true,"left":"\\begin{gather}","right":"\\end{gather}"},{"display":true,"left":"\\begin{CD}","right":"\\end{CD}"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"highlightTag":"em","lunrIndexURL":"/index.json","lunrLanguageCode":"zh","lunrSegmentitURL":"/lib/lunr/lunr.segmentit.js","maxResultLength":10,"noResultsFound":"æ²¡æœ‰æ‰¾åˆ°ç»“æœ","snippetLength":50,"type":"lunr"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
