<!DOCTYPE html>
<html lang="zh-CN">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <title>NGS可视化 - 欢迎来到我的空间🎈</title><meta name="Description" content="欢迎来到griedzx的空间"><meta property="og:url" content="http://griedzx.github.io/ngs_visualization/">
  <meta property="og:site_name" content="欢迎来到我的空间🎈">
  <meta property="og:title" content="NGS可视化">
  <meta property="og:description" content="数据准备 创建合适的文件结构，使用软连接连接对应bam文件 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 #all_bam.txt /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/00.DNase/bam/shoot_DNase-seq.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/01.ATAC_seq/bam/shoot_ATAC-seq.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/H3K9ac_Chip-seq.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/shoot_1_H3K27ac.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/shoot_1_H3K4me3.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/03.methylation/bam/output-prefix.bsmap.mkdup.bam /home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/01.ATAC_seq/bam/Zmays_ear_small.bam /home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/02.Chip_seq/bam/ear_1_H3K27ac.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/02.Chip_seq/bam/ear_1_H3K4me3.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/01.ATAC_seq/bam/ear_ATAC-seq.bam /home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/02.Chip_seq/bam/ear_2_H3K27me3.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/02.Chip_seq/bam/ear_2_H3K4me3.rmdup.bam">
  <meta property="og:locale" content="zh_CN">
  <meta property="og:type" content="article">
    <meta property="article:section" content="posts">
    <meta property="article:published_time" content="2024-03-22T17:19:12+08:00">
    <meta property="article:modified_time" content="2024-03-22T17:19:12+08:00">
    <meta property="og:image" content="http://griedzx.github.io/logo.png">

  <meta name="twitter:card" content="summary_large_image">
  <meta name="twitter:image" content="http://griedzx.github.io/logo.png">
  <meta name="twitter:title" content="NGS可视化">
  <meta name="twitter:description" content="数据准备 创建合适的文件结构，使用软连接连接对应bam文件 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 #all_bam.txt /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/00.DNase/bam/shoot_DNase-seq.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/01.ATAC_seq/bam/shoot_ATAC-seq.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/H3K9ac_Chip-seq.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/shoot_1_H3K27ac.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/shoot_1_H3K4me3.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/03.methylation/bam/output-prefix.bsmap.mkdup.bam /home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/01.ATAC_seq/bam/Zmays_ear_small.bam /home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/02.Chip_seq/bam/ear_1_H3K27ac.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/02.Chip_seq/bam/ear_1_H3K4me3.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/01.ATAC_seq/bam/ear_ATAC-seq.bam /home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/02.Chip_seq/bam/ear_2_H3K27me3.rmdup.bam /home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/02.Chip_seq/bam/ear_2_H3K4me3.rmdup.bam">
      <meta name="twitter:site" content="@xxxx">
<meta name="application-name" content="我的网站">
<meta name="apple-mobile-web-app-title" content="我的网站"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="http://griedzx.github.io/ngs_visualization/" /><link rel="prev" href="http://griedzx.github.io/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" /><link rel="next" href="http://griedzx.github.io/blog_settings/" /><link rel="stylesheet" href="/css/style.min.css"><link rel="preload" href="/lib/fontawesome-free/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"></noscript><link rel="preload" href="/lib/animate/animate.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
        <noscript><link rel="stylesheet" href="/lib/animate/animate.min.css"></noscript><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "NGS可视化",
        "inLanguage": "zh-CN",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "http:\/\/griedzx.github.io\/ngs_visualization\/"
        },"genre": "posts","wordcount":  1241 ,
        "url": "http:\/\/griedzx.github.io\/ngs_visualization\/","datePublished": "2024-03-22T17:19:12+08:00","dateModified": "2024-03-22T17:19:12+08:00","publisher": {
            "@type": "Organization",
            "name": ""},"author": {
                "@type": "Person",
                "name": "griedzx"
            },"description": ""
    }
    </script></head>
    <body data-header-desktop="fixed" data-header-mobile="auto"><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('light' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : 'light' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="欢迎来到我的空间🎈">griedzx 的空间<span class="header-title-post">⚽</span></a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/">🤴 文章 </a><a class="menu-item" href="/tags/"> 标签 </a><a class="menu-item" href="/categories/"> 分类 </a><a class="menu-item" href="https://upptime.daizixi.space/" rel="noopener noreffer" target="_blank"> Uptime-Status </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="搜索">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="清空">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                    <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
                </a></div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="欢迎来到我的空间🎈">griedzx 的空间<span class="header-title-post">⚽</span></a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="搜索">
                            <i class="fas fa-search fa-fw" aria-hidden="true"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="清空">
                            <i class="fas fa-times-circle fa-fw" aria-hidden="true"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin" aria-hidden="true"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        取消
                    </a>
                </div><a class="menu-item" href="/posts/" title="">🤴文章</a><a class="menu-item" href="/tags/" title="">标签</a><a class="menu-item" href="/categories/" title="">分类</a><a class="menu-item" href="https://upptime.daizixi.space/" title="" rel="noopener noreffer" target="_blank">Uptime-Status</a><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                <i class="fas fa-adjust fa-fw" aria-hidden="true"></i>
            </a></div>
    </div>
</header><div class="search-dropdown desktop">
        <div id="search-dropdown-desktop"></div>
    </div>
    <div class="search-dropdown mobile">
        <div id="search-dropdown-mobile"></div>
    </div><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
  <meta name="referrer" content="no-referrer-when-downgrade"><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">目录</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animate__animated animate__flipInX">NGS可视化</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="http://mail.qq.com/cgi-bin/qm_share?t=qm_mailme&amp;email=lOf88frw4aSip6TU5eW69-v5" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw" aria-hidden="true"></i>griedzx</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw" aria-hidden="true"></i>&nbsp;<time datetime="2024-03-22">2024-03-22</time>&nbsp;<i class="fas fa-pencil-alt fa-fw" aria-hidden="true"></i>&nbsp;约 1241 字&nbsp;
                <i class="far fa-clock fa-fw" aria-hidden="true"></i>&nbsp;预计阅读 3 分钟&nbsp;</div>
        </div><div class="details toc" id="toc-static"  data-kept="true">
                <div class="details-summary toc-title">
                    <span>目录</span>
                    <span><i class="details-icon fas fa-angle-right" aria-hidden="true"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents"></nav></div>
            </div><div class="content" id="content"><p>数据准备</p>
<p>创建合适的文件结构，使用软连接连接对应bam文件</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#all_bam.txt</span>
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/00.DNase/bam/shoot_DNase-seq.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/01.ATAC_seq/bam/shoot_ATAC-seq.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/H3K9ac_Chip-seq.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/shoot_1_H3K27ac.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/02.Chip_seq/bam/shoot_1_H3K4me3.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/00.shoot_1/result/03.methylation/bam/output-prefix.bsmap.mkdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/01.ATAC_seq/bam/Zmays_ear_small.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/02.Chip_seq/bam/ear_1_H3K27ac.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/01.ear_1/result/02.Chip_seq/bam/ear_1_H3K4me3.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/01.ATAC_seq/bam/ear_ATAC-seq.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/02.Chip_seq/bam/ear_2_H3K27me3.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/02.Chip_seq/bam/ear_2_H3K4me3.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/02.Chip_seq/bam/ear_2_H3K9ac.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/02_ear_2/result/03.methylation/bam/output-prefix.bsmap.mkdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/03.tassel/result/01.ATAC_seq/bam/tassel_ATAC-seq.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/03.tassel/result/02.Chip_seq/bam/tassel_H3K27me3.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/03.tassel/result/02.Chip_seq/bam/tassel_H3K4me3.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/03.tassel/result/02.Chip_seq/bam/tassel_H3K9ac.rmdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/03.tassel/result/03.methylation/bam/output-prefix.bsmap.mkdup.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/00.DNase/bam/DNase.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/01.ATAC_seq/bam/Zmays_young_leaf_2c.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/02.Chip_seq/bam/H3K9ac_Chip-seq.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/02.Chip_seq/bam/SRR9016266_H3K27ac_ChIP-seq_rep1_picard_.sort.bam.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/02.Chip_seq/bam/SRR9016267_H3K4me1_ChIP-seq_Rep1_picard_.sort.bam.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/02.Chip_seq/bam/SRR9016268_RNA11_ChIP-seq_Rep1_picard_.sort.bam.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/02.Chip_seq/bam/SRR9016269_H3k27me3_ChIP-seq_Rep1_picard_.sort.bam.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/02.Chip_seq/bam/SRR9016270_H3K4me3_ChIP-seq_Rep1_picard_.sort.bam.bam
</span></span><span class="line"><span class="cl">/home/data/yuanh/work_data/230612_encher_data/04.shoot_2/result/03.methylation/bam/shoot_2_methylation.sorted.bam
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="cp">#! /bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span>
</span></span><span class="line"><span class="cl"><span class="k">while</span> <span class="nv">IFS</span><span class="o">=</span> <span class="nb">read</span> -r line<span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">  <span class="c1"># tissue、omic、file_path</span>
</span></span><span class="line"><span class="cl">  <span class="nv">file_path</span><span class="o">=</span><span class="s2">&#34;</span><span class="nv">$line</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">  <span class="nv">omic</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span> <span class="s2">&#34;</span><span class="nv">$file_path</span><span class="s2">&#34;</span> <span class="p">|</span> awk -F<span class="s1">&#39;/&#39;</span> <span class="s1">&#39;{gsub(/[0-9]+\.|_/, &#34;&#34;, $(NF-2)); print $(NF-2)}&#39;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="nv">tissue</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span> <span class="s2">&#34;</span><span class="nv">$file_path</span><span class="s2">&#34;</span> <span class="p">|</span> awk -F<span class="s1">&#39;/&#39;</span> <span class="s1">&#39;{gsub(/[0-9]+\.|_/, &#34;&#34;, $(NF-4)); print $(NF-4)}&#39;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1"># mkdir</span>
</span></span><span class="line"><span class="cl">  <span class="nv">target_dir</span><span class="o">=</span><span class="s2">&#34;./</span><span class="nv">$tissue</span><span class="s2">/</span><span class="nv">$omic</span><span class="s2">/bam&#34;</span>
</span></span><span class="line"><span class="cl">  <span class="nb">echo</span> <span class="s2">&#34;</span><span class="nv">$target_dir</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">  mkdir -p <span class="s2">&#34;</span><span class="nv">$target_dir</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">  <span class="c1"># ln -s</span>
</span></span><span class="line"><span class="cl">  <span class="k">if</span> <span class="o">[</span> <span class="s2">&#34;</span><span class="nv">$omic</span><span class="s2">&#34;</span> <span class="o">==</span> <span class="s2">&#34;Chip_seq&#34;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># Chip_seq</span>
</span></span><span class="line"><span class="cl">    <span class="nv">link_name</span><span class="o">=</span><span class="k">$(</span>basename <span class="s2">&#34;</span><span class="nv">$file_path</span><span class="s2">&#34;</span> <span class="p">|</span> grep -oP <span class="s1">&#39;H[123]K[1-9]*(me[123]|ac)|RNA11&#39;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">  <span class="k">else</span>
</span></span><span class="line"><span class="cl">    <span class="c1">#omic</span>
</span></span><span class="line"><span class="cl">    <span class="nv">link_name</span><span class="o">=</span><span class="s2">&#34;</span><span class="nv">$omic</span><span class="s2">.bam&#34;</span>
</span></span><span class="line"><span class="cl">  <span class="k">fi</span>
</span></span><span class="line"><span class="cl">  ln -s <span class="s2">&#34;</span><span class="nv">$file_path</span><span class="s2">&#34;</span> <span class="s2">&#34;</span><span class="nv">$target_dir</span><span class="s2">/</span><span class="nv">$link_name</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span> &lt; all_bam.txt
</span></span></code></pre></td></tr></table>
</div>
</div><p>递归查询有多少软链接建立，和txt中文件路径相符</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="o">(</span>base<span class="o">)</span> yuanhx@T640server 23:08:13 ~/dzx/ultimate/1.input_circumstance/data
</span></span><span class="line"><span class="cl">$ find ./ -type l <span class="p">|</span>wc -l
</span></span><span class="line"><span class="cl"><span class="m">28</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p><strong>deeptools</strong>去处理链特异性的数据</p>
<p><strong>我们要分析测序数据在基因上的分布特征，显然不能将来自于负链基因数据统计到同一区域的正链基因上。</strong> 一个简单的思路就是：</p>
<ul>
<li>首先将BAM文件根据链拆分成正链和负链的BAM文件；</li>
<li>再用正链的BAM文件去和正链基因的BED文件去 <code>computeMatrix</code>，负链同理；</li>
<li>最后再将正负链 <code>computeMatrix</code>的结果进行合并。</li>
</ul>
<p>bam文件区分正负链</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1"># 选择比对到正链的reads</span>
</span></span><span class="line"><span class="cl">samtools view -F <span class="m">16</span> -b -o positive.bam example.bam
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># 选择比对到负链的reads</span>
</span></span><span class="line"><span class="cl">samtools view -f <span class="m">16</span> -b -o negative.bam example.bam 
</span></span></code></pre></td></tr></table>
</div>
</div><p>&ldquo;-f&quot;参数用于选择具有特定标志的读取，&quot;-F&quot;参数用于排除具有特定标志的读取</p>
<p>SAM/BAM文件中，每个读取都有一个FLAG字段，这个字段是一个位标志，用于表示读取的各种属性，这些属性包括读取是否配对，是否比对到参考序列，是否是配对中的第一读取，等等</p>
<p>对于链特异性，我们关注的是第5位，如果这一位是1，那么读取比对到参考序列的负链，如果是0，那么读取比对到参考序列的正链。这一位的值是16，所以使用16作为参数</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#divide the bam files into positive and negative</span>
</span></span><span class="line"><span class="cl"><span class="nv">bam_files</span><span class="o">=</span><span class="k">$(</span>find ./ -type l<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> bam in <span class="nv">$bam_files</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="c1">#remove the .bam which is the last 4 characters</span>
</span></span><span class="line"><span class="cl">    <span class="nv">file_name</span><span class="o">=</span><span class="si">${</span><span class="nv">bam</span><span class="p">:</span><span class="k">:-</span><span class="nv">4</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">    samtools view -@ <span class="m">9</span> -f <span class="m">16</span> -b -o <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_negative.bam <span class="nv">$bam</span>
</span></span><span class="line"><span class="cl">    samtools view -@ <span class="m">9</span> -F <span class="m">16</span> -b -o <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_positive.bam <span class="nv">$bam</span>
</span></span><span class="line"><span class="cl">    samtools index <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_negative.bam <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_negative.bam.bai
</span></span><span class="line"><span class="cl">    samtools index <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_positive.bam <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_positive.bam.bai
</span></span><span class="line"><span class="cl">    bamCoverage -b <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_negative.bam -o <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_negative.bw --binSize <span class="m">10</span> --normalizeUsing RPKM --scaleFactor <span class="m">1</span> --numberOfProcessors <span class="m">10</span>
</span></span><span class="line"><span class="cl">    bamCoverage -b <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_positive.bam -o <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_positive.bw --binSize <span class="m">10</span> --normalizeUsing RPKM --scaleFactor <span class="m">1</span> --numberOfProcessors <span class="m">10</span>
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>bed基因文件区分正负链</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#divide gene.bed into positive and negative</span>
</span></span><span class="line"><span class="cl"><span class="nv">bed</span><span class="o">=</span><span class="s2">&#34;./gene.bed&#34;</span>
</span></span><span class="line"><span class="cl">awk <span class="s1">&#39;{if($5==&#34;+&#34;) print $0}&#39;</span> <span class="nv">$bed</span> &gt; gene_positive.bed
</span></span><span class="line"><span class="cl">awk <span class="s1">&#39;{if($5==&#34;-&#34;) print $0}&#39;</span> <span class="nv">$bed</span> &gt; gene_negative.bed
</span></span></code></pre></td></tr></table>
</div>
</div><p>分正负链基因计算reads富集分数</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="cp">#! /bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1"># Path: static/NGS_visualization/script</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#sbatch --dependency=afterok:31482 ./run.slurm</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#using computeMatrix to calculate the matrix</span>
</span></span><span class="line"><span class="cl"><span class="nv">bam_files</span><span class="o">=</span><span class="k">$(</span>find ./ -type l<span class="p">|</span> grep <span class="s2">&#34;bam</span>$<span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> bam in <span class="nv">$bam_files</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">file_path</span><span class="o">=</span><span class="si">${</span><span class="nv">dirname</span><span class="p">(dirname(</span><span class="nv">$bam</span><span class="p">))</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">    mkdir -p <span class="nv">$file_path</span>/matrix
</span></span><span class="line"><span class="cl">    <span class="nv">omic</span><span class="o">=</span><span class="si">${</span><span class="nv">basename</span><span class="p">(bam:</span><span class="k">:-</span><span class="nv">4</span><span class="p">)</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">    <span class="nv">file_name</span><span class="o">=</span><span class="si">${</span><span class="nv">bam</span><span class="p">:</span><span class="k">:-</span><span class="nv">4</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">    computeMatrix scale-regions -p <span class="m">12</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -S <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_positive.bw <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -R gene_positive.bed <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -b <span class="m">1000</span> -a <span class="m">1000</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_positive.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --binSize <span class="m">10</span> --missingDataAsZero <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --skipZeros <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --outFileNameMatrix <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_positive.tab
</span></span><span class="line"><span class="cl">    computeMatrix scale-regions -p <span class="m">12</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -S <span class="si">${</span><span class="nv">file_name</span><span class="si">}</span>_negative.bw <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -R gene_negative.bed <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -b <span class="m">1000</span> -a <span class="m">1000</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -o <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_negative.gz
</span></span><span class="line"><span class="cl">      --binSize <span class="m">10</span> --missingDataAsZero <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --skipZeros <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --outFileNameMatrix <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_negative.tab
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>合并正负链tab文件</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"> <span class="c1">#合并正负链tab文件</span>
</span></span><span class="line"><span class="cl"><span class="nv">bam_files</span><span class="o">=</span><span class="k">$(</span>find ./ -type l<span class="p">|</span> grep <span class="s2">&#34;bam</span>$<span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> bam in <span class="nv">$bam_files</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">file_path</span><span class="o">=</span><span class="k">$(</span>dirname <span class="s2">&#34;</span><span class="k">$(</span>dirname <span class="s2">&#34;</span><span class="nv">$bam</span><span class="s2">&#34;</span><span class="k">)</span><span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="nv">omic</span><span class="o">=</span><span class="k">$(</span>basename <span class="s2">&#34;</span><span class="si">${</span><span class="nv">bam</span><span class="p">:</span><span class="k">:-</span><span class="nv">4</span><span class="si">}</span><span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="c1">#根据两个文件表头确定新表头</span>
</span></span><span class="line"><span class="cl">    <span class="c1">#first line</span>
</span></span><span class="line"><span class="cl">    <span class="nv">group_boundaries_positive</span><span class="o">=</span><span class="k">$(</span>zcat <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_positive.gz <span class="p">|</span> head -n <span class="m">1</span> <span class="p">|</span> grep -oP <span class="s1">&#39;&#34;group_boundaries&#34;:\[\K[^\]]*&#39;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="nv">group_boundaries_negative</span><span class="o">=</span><span class="k">$(</span>zcat <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_negative.gz <span class="p">|</span> head -n <span class="m">1</span> <span class="p">|</span> grep -oP <span class="s1">&#39;&#34;group_boundaries&#34;:\[\K[^\]]*&#39;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># 计算新的 group_boundaries 的值</span>
</span></span><span class="line"><span class="cl">    <span class="nv">num1</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span> <span class="nv">$group_boundaries_positive</span> <span class="p">|</span> awk -F <span class="s1">&#39;,&#39;</span> <span class="s1">&#39;{print $2}&#39;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="nv">num2</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span> <span class="nv">$group_boundaries_negative</span> <span class="p">|</span> awk -F <span class="s1">&#39;,&#39;</span> <span class="s1">&#39;{print $2}&#39;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="nv">new_group_boundaries</span><span class="o">=</span><span class="s2">&#34;0,</span><span class="k">$((</span>num1+num2<span class="k">))</span><span class="s2">&#34;</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># 从 positive 文件中提取 JSON 数据，并将 group_boundaries 和 sample_labels 的值进行修改</span>
</span></span><span class="line"><span class="cl">    <span class="nv">new_json</span><span class="o">=</span><span class="k">$(</span>zcat <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_positive.gz <span class="p">|</span> head -n <span class="m">1</span> <span class="p">|</span> sed <span class="s2">&#34;s/\&#34;group_boundaries\&#34;:\[[^]]*\]/\&#34;group_boundaries\&#34;:[</span><span class="nv">$new_group_boundaries</span><span class="s2">]/g&#34;</span> <span class="p">|</span> sed <span class="s2">&#34;s/\&#34;sample_labels\&#34;:\[[^]]*\]/\&#34;sample_labels\&#34;:[\&#34;</span><span class="nv">$omic</span><span class="s2">\&#34;]/g&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="c1"># 输出新的 JSON 数据</span>
</span></span><span class="line"><span class="cl">    <span class="nb">echo</span> <span class="nv">$new_json</span> &gt; <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>.tab
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">    <span class="c1">#每个gz文件去除第一行行合并</span>
</span></span><span class="line"><span class="cl">    zcat <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_positive.gz <span class="p">|</span> tail -n +2 &gt;&gt; <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>.tab
</span></span><span class="line"><span class="cl">    zcat <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>_negative.gz <span class="p">|</span> tail -n +2 &gt;&gt; <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>.tab
</span></span><span class="line"><span class="cl">    <span class="c1">#gzip the tab file</span>
</span></span><span class="line"><span class="cl">    gzip -c <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>.tab &gt; <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>.tab.gz
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span></code></pre></td></tr></table>
</div>
</div><p>画图</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-shell" data-lang="shell"><span class="line"><span class="cl"><span class="c1">#plotHeatmap</span>
</span></span><span class="line"><span class="cl"><span class="c1">#! /bin/bash</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">bam_files</span><span class="o">=</span><span class="k">$(</span>find ./ -type l<span class="p">|</span> grep <span class="s2">&#34;bam</span>$<span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> bam in <span class="nv">$bam_files</span><span class="p">;</span> <span class="k">do</span>
</span></span><span class="line"><span class="cl">    <span class="nv">file_path</span><span class="o">=</span><span class="k">$(</span>dirname <span class="s2">&#34;</span><span class="k">$(</span>dirname <span class="s2">&#34;</span><span class="nv">$bam</span><span class="s2">&#34;</span><span class="k">)</span><span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    <span class="nv">omic</span><span class="o">=</span><span class="k">$(</span>basename <span class="s2">&#34;</span><span class="si">${</span><span class="nv">bam</span><span class="p">:</span><span class="k">:-</span><span class="nv">4</span><span class="si">}</span><span class="s2">&#34;</span><span class="k">)</span>
</span></span><span class="line"><span class="cl">    mkdir -p <span class="nv">$file_path</span>/plotheatmap
</span></span><span class="line"><span class="cl">    plotHeatmap -m <span class="nv">$file_path</span>/matrix/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>.tab.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      -out <span class="nv">$file_path</span>/plotheatmap/<span class="si">${</span><span class="nv">omic</span><span class="si">}</span>.pdf <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --dpi <span class="m">300</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --colorMap YlGnBu  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --missingDataColor <span class="s2">&#34;#FFF6EB&#34;</span><span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --heatmapHeight <span class="m">21</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --startLabel <span class="s2">&#34;TSS&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --endLabel <span class="s2">&#34;TTS&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --regionsLabel <span class="s2">&#34;gene&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --legendLocation <span class="s2">&#34;none&#34;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>      --plotTitle <span class="nv">$omic</span> 
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span></code></pre></td></tr></table>
</div>
</div></div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>更新于 2024-03-22</span>
            </div></div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/ngs_visualization/index.md" target="_blank">阅读原始文档</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="分享到 Twitter" data-sharer="twitter" data-url="http://griedzx.github.io/ngs_visualization/" data-title="NGS可视化" data-via="xxxx"><i class="fab fa-twitter fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 Facebook" data-sharer="facebook" data-url="http://griedzx.github.io/ngs_visualization/"><i class="fab fa-facebook-square fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 Hacker News" data-sharer="hackernews" data-url="http://griedzx.github.io/ngs_visualization/" data-title="NGS可视化"><i class="fab fa-hacker-news fa-fw" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 Line" data-sharer="line" data-url="http://griedzx.github.io/ngs_visualization/" data-title="NGS可视化"><i data-svg-src="/lib/simple-icons/icons/line.min.svg" aria-hidden="true"></i></a><a href="javascript:void(0);" title="分享到 微博" data-sharer="weibo" data-url="http://griedzx.github.io/ngs_visualization/" data-title="NGS可视化"><i class="fab fa-weibo fa-fw" aria-hidden="true"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">返回</a></span>&nbsp;|&nbsp;<span><a href="/">主页</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/%E7%94%9F%E4%BF%A1%E8%BF%9B%E5%B1%95-atac%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86/" class="prev" rel="prev" title="生信进展-ATAC数据处理"><i class="fas fa-angle-left fa-fw" aria-hidden="true"></i>生信进展-ATAC数据处理</a>
            <a href="/blog_settings/" class="next" rel="next" title="博客设置">博客设置<i class="fas fa-angle-right fa-fw" aria-hidden="true"></i></a></div>
</div>
<div id="comments"></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">由 <a href="https://gohugo.io/" target="_blank" rel="noopener noreffer" title="Hugo 0.132.0">Hugo</a> 强力驱动 | 主题 - <a href="https://github.com/dillonzq/LoveIt" target="_blank" rel="noopener noreffer" title="LoveIt 0.2.11"><i class="far fa-kiss-wink-heart fa-fw" aria-hidden="true"></i> LoveIt</a>
                </div><div class="footer-line" itemscope itemtype="http://schema.org/CreativeWork"><i class="far fa-copyright fa-fw" aria-hidden="true"></i><span itemprop="copyrightYear">2023 - 2024</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="http://mail.qq.com/cgi-bin/qm_share?t=qm_mailme&amp;email=lOf88frw4aSip6TU5eW69-v5" target="_blank">griedzx</a></span>&nbsp;|&nbsp;<span class="license"><a rel="license external nofollow noopener noreffer" href="https://creativecommons.org/licenses/by-nc/4.0/" target="_blank">CC BY-NC 4.0</a></span></div>
        </div>

        
        <div class="busuanzi-footer">
        <span id="busuanzi_container_site_pv">
            本站总访问量<span id="busuanzi_value_site_pv"></span>次
        </span>
        <span id="busuanzi_container_site_uv">
            本站访客数<span id="busuanzi_value_site_uv"></span>人次
        </span>
        </div></footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="回到顶部">
                <i class="fas fa-arrow-up fa-fw" aria-hidden="true"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="查看评论">
                <i class="fas fa-comment fa-fw" aria-hidden="true"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/cookieconsent/cookieconsent.min.css"><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.stemmer.support.min.js"></script><script type="text/javascript" src="/lib/lunr/lunr.zh.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/contrib/mhchem.min.js"></script><script type="text/javascript" src="/lib/cookieconsent/cookieconsent.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"复制到剪贴板","maxShownLines":50},"comment":{},"cookieconsent":{"content":{"dismiss":"同意","link":"了解更多","message":"本网站使用 Cookies 来改善您的浏览体验."},"enable":true,"palette":{"button":{"background":"#f0f0f0"},"popup":{"background":"#1aa3ff"}},"theme":"edgeless"},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":true,"left":"\\begin{equation}","right":"\\end{equation}"},{"display":true,"left":"\\begin{equation*}","right":"\\end{equation*}"},{"display":true,"left":"\\begin{align}","right":"\\end{align}"},{"display":true,"left":"\\begin{align*}","right":"\\end{align*}"},{"display":true,"left":"\\begin{alignat}","right":"\\end{alignat}"},{"display":true,"left":"\\begin{alignat*}","right":"\\end{alignat*}"},{"display":true,"left":"\\begin{gather}","right":"\\end{gather}"},{"display":true,"left":"\\begin{CD}","right":"\\end{CD}"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"highlightTag":"em","lunrIndexURL":"/index.json","lunrLanguageCode":"zh","lunrSegmentitURL":"/lib/lunr/lunr.segmentit.js","maxResultLength":10,"noResultsFound":"没有找到结果","snippetLength":50,"type":"lunr"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
